<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>EWCE plot — ewce_plot • MultiEWCE</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="EWCE plot — ewce_plot"><meta property="og:description" content="A shallow wrapper for the ewce_plot function."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MultiEWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MultiEWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/MultiEWCE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>EWCE plot</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MultiEWCE/blob/HEAD/R/ewce_plot.R" class="external-link"><code>R/ewce_plot.R</code></a></small>
    <div class="hidden name"><code>ewce_plot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A shallow wrapper for the <a href="https://rdrr.io/pkg/EWCE/man/ewce_plot.html" class="external-link">ewce_plot</a> function.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">ewce_plot</span><span class="op">(</span><span class="va">total_res</span>, mtc_method <span class="op">=</span> <span class="st">"bonferroni"</span>, ctd <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>total_res</dt>
<dd><p>Results data.frame generated using
<a href="https://rdrr.io/pkg/EWCE/man/bootstrap_enrichment_test.html" class="external-link">bootstrap_enrichment_test</a> or
<a href="https://rdrr.io/pkg/EWCE/man/ewce_expression_data.html" class="external-link">ewce_expression_data</a> functions.
Multiple results tables can be
merged into one results table, as long as the 'list' column is set to
distinguish them.
Multiple testing correction is then applied across all merged results.</p></dd>


<dt>mtc_method</dt>
<dd><p>Method to be used for multiple testing correction.
Argument is passed to <a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a> (DEFAULT: "bonferroni).</p></dd>


<dt>ctd</dt>
<dd><p>CellTypeDataset object.
Should be provided so that the dendrogram can be taken from it
and added to plots.</p></dd>


<dt>...</dt>
<dd><p>Arguments passed on to <code><a href="https://rdrr.io/pkg/EWCE/man/ewce_plot.html" class="external-link">EWCE::ewce_plot</a></code></p><dl><dt><code>q_threshold</code></dt>
<dd><p>Corrected significance threshold.</p></dd>

    <dt><code>annotLevel</code></dt>
<dd><p>An integer indicating which level of <code>ctd</code> to
analyse (<em>Default: 1</em>).</p></dd>

    <dt><code>heights</code></dt>
<dd><p>The relative heights row in the grid. 
Will get repeated to match the dimensions of the grid.
Passed to <a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a>.</p></dd>

    <dt><code>make_dendro</code></dt>
<dd><p>Add a dendrogram (requires <code>ctd</code>).</p></dd>

    <dt><code>verbose</code></dt>
<dd><p>Print messages.</p></dd>

  
</dl></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A bar chart with dendrogram of EWCE results in each cell type.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">full_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EWCE/man/example_bootstrap_results.html" class="external-link">example_bootstrap_results</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Recomputing example bootstrap results.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 core(s) assigned as workers (1 reserved).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Generating gene background for mouse x human ==&gt; human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering ortholog reports.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 21,207 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 21,207 genes from mouse.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.frame format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from Gene.Symbol.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 21,207 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting mouse ==&gt; human orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17,355 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17,355 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 131 genes that have multiple input_gene per ortholog_gene (many:1).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 498 genes that have multiple ortholog_gene per input_gene (1:many).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    4,725 / 21,207 (22%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    16,482 / 21,207 (78%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering ortholog reports.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 intersect background genes used.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising CellTypeDataset</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking gene list inputs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running without gene size control.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17 hit gene(s) remain after filtering.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing gene scores.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using previously sampled genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing gene counts.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for enrichment in 7 cell types...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sorting results by p-value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing BH-corrected q-values.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 significant cell type enrichment results @ q&lt;0.05 : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    CellType annotLevel p fold_change sd_from_mean q</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 microglia          1 0    2.042317     4.460971 0</span>
<span class="r-in"><span><span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu">MultiEWCE</span><span class="fu">::</span><span class="fu">ewce_plot</span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">full_results</span><span class="op">$</span><span class="va">results</span><span class="op">)</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robert Gordon-Smith, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

