<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Generate results — gen_results • MSTExplorer</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Generate results — gen_results"><meta property="og:description" content="Generates EWCE results on multiple gene lists in parallel by calling
ewce_para. It allows you to stop the analysis and then continue later
from where you left off as it checks the results output directory for
finished gene lists and removes them from the input.
It also excludes gene lists with
less than 4 unique genes (which cause errors in EWCE analysis)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MSTExplorer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MSTExplorer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/MSTExplorer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate results</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MSTExplorer/blob/master/R/gen_results.R" class="external-link"><code>R/gen_results.R</code></a></small>
    <div class="hidden name"><code>gen_results.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Generates EWCE results on multiple gene lists in parallel by calling
<code>ewce_para</code>. It allows you to stop the analysis and then continue later
from where you left off as it checks the results output directory for
finished gene lists and removes them from the input.
It also excludes gene lists with
less than 4 unique genes (which cause errors in EWCE analysis).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">gen_results</span><span class="op">(</span></span>
<span>  <span class="va">ctd</span>,</span>
<span>  <span class="va">gene_data</span>,</span>
<span>  list_name_column <span class="op">=</span> <span class="st">"hpo_id"</span>,</span>
<span>  gene_column <span class="op">=</span> <span class="st">"gene_symbol"</span>,</span>
<span>  list_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">gene_data</span><span class="op">[[</span><span class="va">list_name_column</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span>
<span>  reps <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  annotLevel <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  sctSpecies <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  force_new <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  bg <span class="op">=</span> <span class="fu"><a href="get_bg.html">get_bg</a></span><span class="op">(</span>species1 <span class="op">=</span> <span class="va">genelistSpecies</span>, species2 <span class="op">=</span> <span class="va">sctSpecies</span>, overwrite <span class="op">=</span> <span class="va">force_new</span><span class="op">)</span>,</span>
<span>  min_genes <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  parallel_boot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  save_dir_tmp <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  save_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  verbose <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-ctd">ctd<a class="anchor" aria-label="anchor" href="#arg-ctd"></a></dt>
<dd><p>CellTypeDataset generated using
<a href="https://rdrr.io/pkg/EWCE/man/generate_celltype_data.html" class="external-link">generate_celltype_data</a>.</p></dd>


<dt id="arg-gene-data">gene_data<a class="anchor" aria-label="anchor" href="#arg-gene-data"></a></dt>
<dd><p>data frame of gene list names and genes
(see <a href="https://rdrr.io/pkg/HPOExplorer/man/get_gene_lists.html" class="external-link">get_gene_lists</a>).</p></dd>


<dt id="arg-list-name-column">list_name_column<a class="anchor" aria-label="anchor" href="#arg-list-name-column"></a></dt>
<dd><p>The name of the gene_data column
that has the gene list names.</p></dd>


<dt id="arg-gene-column">gene_column<a class="anchor" aria-label="anchor" href="#arg-gene-column"></a></dt>
<dd><p>The name of the gene_data column that contains the genes.</p></dd>


<dt id="arg-list-names">list_names<a class="anchor" aria-label="anchor" href="#arg-list-names"></a></dt>
<dd><p>character vector of gene list names.</p></dd>


<dt id="arg-reps">reps<a class="anchor" aria-label="anchor" href="#arg-reps"></a></dt>
<dd><p>Number of random gene lists to generate (<em>Default: 100</em>,
but should be &gt;=10,000 for publication-quality results).</p></dd>


<dt id="arg-annotlevel">annotLevel<a class="anchor" aria-label="anchor" href="#arg-annotlevel"></a></dt>
<dd><p>An integer indicating which level of <code>sct_data</code> to
analyse (<em>Default: 1</em>).</p></dd>


<dt id="arg-genelistspecies">genelistSpecies<a class="anchor" aria-label="anchor" href="#arg-genelistspecies"></a></dt>
<dd><p>Species that <code>hits</code> genes came from
(no longer limited to just "mouse" and "human").
See <a href="https://rdrr.io/pkg/EWCE/man/list_species.html" class="external-link">list_species</a> for all available species.</p></dd>


<dt id="arg-sctspecies">sctSpecies<a class="anchor" aria-label="anchor" href="#arg-sctspecies"></a></dt>
<dd><p>Species that <code>sct_data</code> is currently formatted as
(no longer limited to just "mouse" and "human").
See <a href="https://rdrr.io/pkg/EWCE/man/list_species.html" class="external-link">list_species</a> for all available species.</p></dd>


<dt id="arg-force-new">force_new<a class="anchor" aria-label="anchor" href="#arg-force-new"></a></dt>
<dd><p>Overwrite previous results
in the <code>save_dir_tmp</code>.</p></dd>


<dt id="arg-bg">bg<a class="anchor" aria-label="anchor" href="#arg-bg"></a></dt>
<dd><p>List of gene symbols containing the background gene list
(including hit genes). If <code>bg=NULL</code>,
 an appropriate gene background will be created automatically.</p></dd>


<dt id="arg-min-genes">min_genes<a class="anchor" aria-label="anchor" href="#arg-min-genes"></a></dt>
<dd><p>Minimum number of genes per list (default: 4)</p></dd>


<dt id="arg-cores">cores<a class="anchor" aria-label="anchor" href="#arg-cores"></a></dt>
<dd><p>The number of cores to run in parallel (e.g. 8) <code>int</code>.</p></dd>


<dt id="arg-parallel-boot">parallel_boot<a class="anchor" aria-label="anchor" href="#arg-parallel-boot"></a></dt>
<dd><p>Parallelise at the level of bootstrap iterations,
rather than across gene lists.</p></dd>


<dt id="arg-save-dir-tmp">save_dir_tmp<a class="anchor" aria-label="anchor" href="#arg-save-dir-tmp"></a></dt>
<dd><p>Folder to save intermediate results files to
(one file per gene list). Set to <code>NULL</code> to skip saving temporary files.</p></dd>


<dt id="arg-save-dir">save_dir<a class="anchor" aria-label="anchor" href="#arg-save-dir"></a></dt>
<dd><p>Folder to save merged results in.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Print messages.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Arguments passed on to <code><a href="https://rdrr.io/pkg/EWCE/man/bootstrap_enrichment_test.html" class="external-link">EWCE::bootstrap_enrichment_test</a></code></p><dl><dt><code>sct_data</code></dt>
<dd><p>List generated using <a href="https://rdrr.io/pkg/EWCE/man/generate_celltype_data.html" class="external-link">generate_celltype_data</a>.</p></dd>

    <dt><code>hits</code></dt>
<dd><p>List of gene symbols containing the target gene list.
Will automatically be converted to human gene symbols
if <code>geneSizeControl=TRUE</code>.</p></dd>

    <dt><code>sctSpecies_origin</code></dt>
<dd><p>Species that the <code>sct_data</code>
originally came from, regardless of its current gene format
(e.g. it was previously converted from mouse to human gene orthologs).
 This is used for computing an appropriate backgrund.</p></dd>

    <dt><code>output_species</code></dt>
<dd><p>Species to convert <code>sct_data</code> and <code>hits</code> to
(Default: "human").
See <a href="https://rdrr.io/pkg/EWCE/man/list_species.html" class="external-link">list_species</a> for all available species.</p></dd>

    <dt><code>method</code></dt>
<dd><p>R package to use for gene mapping:</p><ul><li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives consensus scores for each gene mapping based on a
 several different data sources.</p></li>
</ul></dd>

    <dt><code>no_cores</code></dt>
<dd><p>Number of cores to parallelise
bootstrapping <code>reps</code> over.</p></dd>

    <dt><code>geneSizeControl</code></dt>
<dd><p>Whether you want to control for
GC content and transcript length. Recommended if the gene list originates
from genetic studies (<em>Default: FALSE</em>).
If set to <code>TRUE</code>, then <code>hits</code> must be from humans.</p></dd>

    <dt><code>controlledCT</code></dt>
<dd><p>[Optional] If not NULL, and instead is the name of a
cell type, then the bootstrapping controls for expression within that
cell type.</p></dd>

    <dt><code>mtc_method</code></dt>
<dd><p>Multiple-testing correction method
(passed to <a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a>).</p></dd>

    <dt><code>sort_results</code></dt>
<dd><p>Sort enrichment results from
smallest to largest p-values.</p></dd>

    <dt><code>standardise_sct_data</code></dt>
<dd><p>Should <code>sct_data</code> be standardised?
if <code>TRUE</code>:</p><ul><li><p>When <code>sctSpecies!=output_species</code>
 the <code>sct_data</code> will be checked for object formatting and
 the genes will be converted to the orthologs of the <code>output_species</code>
 with <a href="https://rdrr.io/pkg/EWCE/man/standardise_ctd.html" class="external-link">standardise_ctd</a>
  (which calls <a href="https://rdrr.io/pkg/orthogene/man/map_genes.html" class="external-link">map_genes</a> internally).</p></li>
<li><p>When <code>sctSpecies==output_species</code>,
 the <code>sct_data</code> will be checked for object formatting
 with  <a href="https://rdrr.io/pkg/EWCE/man/standardise_ctd.html" class="external-link">standardise_ctd</a>, but the gene names
 will remain untouched.</p></li>
</ul></dd>

    <dt><code>standardise_hits</code></dt>
<dd><p>Should <code>hits</code> be standardised?
If <code>TRUE</code>:</p><ul><li><p>When <code>genelistSpecies!=output_species</code>,
 the genes will be converted to the orthologs of the <code>output_species</code>
 with <a href="https://rdrr.io/pkg/orthogene/man/convert_orthologs.html" class="external-link">convert_orthologs</a>.</p></li>
<li><p>When <code>genelistSpecies==output_species</code>,
 the genes will be standardised with <a href="https://rdrr.io/pkg/orthogene/man/map_genes.html" class="external-link">map_genes</a>.</p></li>
</ul><p>If <code>FALSE</code>, <code>hits</code> will be passed on to subsequent steps as-is.</p></dd>

    <dt><code>localHub</code></dt>
<dd><p>If working offline, add argument localHub=TRUE to work
with a local, non-updated hub; It will only have resources available that
have previously been downloaded. If offline, Please also see BiocManager
vignette section on offline use to ensure proper functionality.</p></dd>

    <dt><code>store_gene_data</code></dt>
<dd><p>Store sampled gene data for every bootstrap iteration.
When the number of bootstrap <code>reps</code> is very high (&gt;=100k) and/or
 the number of genes in <code>hits</code> is very high, you may want
 to set <code>store_gene_data=FALSE</code> to avoid using excessive amounts of
 CPU memory.</p></dd>


</dl></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>All results as a dataframe.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The gene_data should be a data frame that contains a column of gene
list names (e.g. the column may be called "hpo_name"), and a column of
genes (e.g. "gene_symbol"). For example:</p><table class="table table"><tr><td>hpo_name</td><td>gene_symbol</td></tr><tr><td>"Abnormal heart"</td><td>gene X</td></tr><tr><td>"Abnormal heart"</td><td>gene Y</td></tr><tr><td>"Poor vision"</td><td>gene Z</td></tr><tr><td>"Poor vision"</td><td>gene Y</td></tr><tr><td>etc...</td><td></td></tr></table><p>For more information on this see docs for get_gene_list
(<a href="https://rdrr.io/pkg/HPOExplorer/man/get_gene_lists.html" class="external-link">get_gene_lists</a>).</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">gene_data</span> <span class="op">&lt;-</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/load_phenotype_to_genes.html" class="external-link">load_phenotype_to_genes</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2025-05-06</span>
<span class="r-in"><span><span class="va">list_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">gene_data</span><span class="op">$</span><span class="va">hpo_id</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="load_example_CTD.html">load_example_ctd</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading ctd_DescartesHuman_example.rds</span>
<span class="r-in"><span><span class="va">all_results</span> <span class="op">&lt;-</span> <span class="fu">gen_results</span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,</span></span>
<span class="r-in"><span>                           gene_data <span class="op">=</span> <span class="va">gene_data</span>,</span></span>
<span class="r-in"><span>                           list_names <span class="op">=</span> <span class="va">list_names</span>,</span></span>
<span class="r-in"><span>                           reps <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Validating gene lists..</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2 / 5 gene lists are valid.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Useing cached bg.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: 2025-08-08</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Background contains 78,687 genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing gene counts.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing gene counts.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done in: 3.2 seconds.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving results ==&gt; /tmp/RtmpsW0tFt/gen_results.rds</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Robert Gordon-Smith, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

