<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Get background genes — get_bg • MSTExplorer</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Get background genes — get_bg"><meta property="og:description" content="Generate background genes given one or more species.
Caches the list to avoid excessive API calls to
g:Profiler."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MSTExplorer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MSTExplorer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/MSTExplorer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get background genes</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MSTExplorer/blob/master/R/get_bg.R" class="external-link"><code>R/get_bg.R</code></a></small>
    <div class="hidden name"><code>get_bg.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Generate background genes given one or more species.
Caches the list to avoid excessive API calls to
<a href="https://biit.cs.ut.ee/gprofiler" class="external-link">g:Profiler</a>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_bg</span><span class="op">(</span></span>
<span>  species1 <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  species2 <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"gprofiler"</span>,</span>
<span>  save_dir <span class="op">=</span> <span class="fu">KGExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/KGExplorer/man/cache_.html" class="external-link">cache_dir</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"MSTExplorer"</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-species-">species1<a class="anchor" aria-label="anchor" href="#arg-species-"></a></dt>
<dd><p>First species.</p></dd>


<dt id="arg-species-">species2<a class="anchor" aria-label="anchor" href="#arg-species-"></a></dt>
<dd><p>Second species.</p></dd>


<dt id="arg-method">method<a class="anchor" aria-label="anchor" href="#arg-method"></a></dt>
<dd><p>R package to use for gene mapping:</p><ul><li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives consensus scores for each gene mapping based on a
 several different data sources.</p></li>
</ul></dd>


<dt id="arg-save-dir">save_dir<a class="anchor" aria-label="anchor" href="#arg-save-dir"></a></dt>
<dd><p>Directory to save data to.</p></dd>


<dt id="arg-overwrite">overwrite<a class="anchor" aria-label="anchor" href="#arg-overwrite"></a></dt>
<dd><p>Should any local files of the same name be overwritten?
default <code>TRUE</code>.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Print messages.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Arguments passed on to <code><a href="https://rdrr.io/pkg/orthogene/man/create_background.html" class="external-link">orthogene::create_background</a></code></p><dl><dt><code>output_species</code></dt>
<dd><p>Species to convert all genes from
<code>species1</code> and <code>species2</code> to first.
 <code>Default="human"</code>, but can be to either any species
  supported by <span class="pkg">orthogene</span>, including
 <code>species1</code> or <code>species2</code>.</p></dd>

    <dt><code>as_output_species</code></dt>
<dd><p>Return background gene list as
<code>output_species</code> orthologs, instead of the
 gene names of the original input species.</p></dd>

    <dt><code>use_intersect</code></dt>
<dd><p>When <code>species1</code> and <code>species2</code> are both
different from <code>output_species</code>, this argument will determine whether
to use the intersect (<code>TRUE</code>) or union (<code>FALSE</code>) of all genes
from <code>species1</code> and <code>species2</code>.</p></dd>

    <dt><code>bg</code></dt>
<dd><p>User supplied background list that will be returned to the
user after removing duplicate genes.</p></dd>

    <dt><code>gene_map</code></dt>
<dd><p>User-supplied <code>gene_map</code> data table from
<a href="https://rdrr.io/pkg/orthogene/man/map_orthologs.html" class="external-link">map_orthologs</a> or <a href="https://rdrr.io/pkg/orthogene/man/map_genes.html" class="external-link">map_genes</a>.</p></dd>

    <dt><code>non121_strategy</code></dt>
<dd><p>How to handle genes that don't have
1:1 mappings between <code>input_species</code>:<code>output_species</code>.
Options include:<br></p><ul><li><p><code>"drop_both_species" or "dbs" or 1</code> : <br>
Drop genes that have duplicate
mappings in either the <code>input_species</code> or <code>output_species</code> <br>
(<em>DEFAULT</em>).</p></li>
<li><p><code>"drop_input_species" or "dis" or 2</code> : <br>
Only drop genes that have duplicate
mappings in the <code>input_species</code>.</p></li>
<li><p><code>"drop_output_species" or "dos" or 3</code> : <br>
Only drop genes that have duplicate
mappings in the <code>output_species</code>.</p></li>
<li><p><code>"keep_both_species" or "kbs" or 4</code> : <br>
Keep all genes regardless of whether
they have duplicate mappings in either species.</p></li>
<li><p><code>"keep_popular" or "kp" or 5</code> : <br>
Return only the most "popular" interspecies ortholog mappings.
 This procedure tends to yield a greater number of returned genes
 but at the cost of many of them not being true biological 1:1 orthologs.</p></li>
<li><p><code>"sum","mean","median","min" or "max"</code> : <br>
 When <code>gene_df</code> is a matrix and <code>gene_output="rownames"</code>,
  these options will aggregate many-to-one gene mappings
  (<code>input_species</code>-to-<code>output_species</code>)
  after dropping any duplicate genes in the <code>output_species</code>.</p></li>
</ul></dd>


</dl></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A vector of background genes.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">bg</span> <span class="op">&lt;-</span> <span class="fu">get_bg</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Useing cached bg.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: 2025-08-04</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Robert Gordon-Smith, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

