<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot differential outcomes — plot_differential_outcomes • MSTExplorer</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot differential outcomes — plot_differential_outcomes"><meta property="og:description" content="Plot differential outcomes (annotations) of different diseases as they occur
in different cell types via particular phenotypes."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MSTExplorer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MSTExplorer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/MSTExplorer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot differential outcomes</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MSTExplorer/blob/master/R/plot_differential_outcomes.R" class="external-link"><code>R/plot_differential_outcomes.R</code></a></small>
    <div class="hidden name"><code>plot_differential_outcomes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Plot differential outcomes (annotations) of different diseases as they occur
in different cell types via particular phenotypes.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plot_differential_outcomes</span><span class="op">(</span></span>
<span>  <span class="va">results</span>,</span>
<span>  filters <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  facet_var <span class="op">=</span> <span class="st">"disease_name"</span>,</span>
<span>  x_var <span class="op">=</span> <span class="st">"celltype_symptom"</span>,</span>
<span>  y_var <span class="op">=</span> <span class="st">"severity_score_gpt"</span>,</span>
<span>  remove_facets <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  run_stats <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  max_facets <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  q_threshold <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>summary <span class="op">=</span> <span class="fl">0.05</span>, pairwise <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-results">results<a class="anchor" aria-label="anchor" href="#arg-results"></a></dt>
<dd><p>The cell type-phenotype enrichment results generated by
<a href="gen_results.html">gen_results</a>
and merged together with <a href="merge_results.html">merge_results</a></p></dd>


<dt id="arg-filters">filters<a class="anchor" aria-label="anchor" href="#arg-filters"></a></dt>
<dd><p>A named list, where each element in the list is the name of
a column in the data, and the vector within each element represents the
values to include in the final data.</p></dd>


<dt id="arg-facet-var">facet_var<a class="anchor" aria-label="anchor" href="#arg-facet-var"></a></dt>
<dd><p>Variable to facet by.</p></dd>


<dt id="arg-x-var">x_var<a class="anchor" aria-label="anchor" href="#arg-x-var"></a></dt>
<dd><p>Variable to plot on the x-axis.</p></dd>


<dt id="arg-y-var">y_var<a class="anchor" aria-label="anchor" href="#arg-y-var"></a></dt>
<dd><p>Variable to plot on the y-axis.</p></dd>


<dt id="arg-remove-facets">remove_facets<a class="anchor" aria-label="anchor" href="#arg-remove-facets"></a></dt>
<dd><p>A character vector of facets to remove.</p></dd>


<dt id="arg-run-stats">run_stats<a class="anchor" aria-label="anchor" href="#arg-run-stats"></a></dt>
<dd><p>If <code>TRUE</code>, run statistical tests and display the
summary statistics directly on the plot.</p></dd>


<dt id="arg-max-facets">max_facets<a class="anchor" aria-label="anchor" href="#arg-max-facets"></a></dt>
<dd><p>Maximum number of facets to display.</p></dd>


<dt id="arg-q-threshold">q_threshold<a class="anchor" aria-label="anchor" href="#arg-q-threshold"></a></dt>
<dd><p>A list of thresholds for significance testing.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Arguments passed on to <code><a href="https://indrajeetpatil.github.io/ggstatsplot/reference/ggbetweenstats.html" class="external-link">ggstatsplot::ggbetweenstats</a></code></p><dl><dt><code>data</code></dt>
<dd><p>A data frame (or a tibble) from which variables specified are to
be taken. Other data types (e.g., matrix,table, array, etc.) will <strong>not</strong>
be accepted. Additionally, grouped data frames from <code>{dplyr}</code> should be
ungrouped before they are entered as <code>data</code>.</p></dd>

    <dt><code>x</code></dt>
<dd><p>The grouping (or independent) variable from <code>data</code>. In case of a
repeated measures or within-subjects design, if <code>subject.id</code> argument is
not available or not explicitly specified, the function assumes that the
data has already been sorted by such an id by the user and creates an
internal identifier. So if your data is <strong>not</strong> sorted, the results <em>can</em>
be inaccurate when there are more than two levels in <code>x</code> and there are
<code>NA</code>s present. The data is expected to be sorted by user in
subject-1, subject-2, ..., pattern.</p></dd>

    <dt><code>y</code></dt>
<dd><p>The response (or outcome or dependent) variable from <code>data</code>.</p></dd>

    <dt><code>type</code></dt>
<dd><p>A character specifying the type of statistical approach:</p><ul><li><p><code>"parametric"</code></p></li>
<li><p><code>"nonparametric"</code></p></li>
<li><p><code>"robust"</code></p></li>
<li><p><code>"bayes"</code></p></li>
</ul><p>You can specify just the initial letter.</p></dd>

    <dt><code>pairwise.display</code></dt>
<dd><p>Decides <em>which</em> pairwise comparisons to display.
Available options are:</p><ul><li><p><code>"significant"</code> (abbreviation accepted: <code>"s"</code>)</p></li>
<li><p><code>"non-significant"</code> (abbreviation accepted: <code>"ns"</code>)</p></li>
<li><p><code>"all"</code></p></li>
</ul><p>You can use this argument to make sure that your plot is not uber-cluttered
when you have multiple groups being compared and scores of pairwise
comparisons being displayed. If set to <code>"none"</code>, no pairwise comparisons
will be displayed.</p></dd>

    <dt><code>p.adjust.method</code></dt>
<dd><p>Adjustment method for <em>p</em>-values for multiple
comparisons. Possible methods are: <code>"holm"</code> (default), <code>"hochberg"</code>,
<code>"hommel"</code>, <code>"bonferroni"</code>, <code>"BH"</code>, <code>"BY"</code>, <code>"fdr"</code>, <code>"none"</code>.</p></dd>

    <dt><code>effsize.type</code></dt>
<dd><p>Type of effect size needed for <em>parametric</em> tests. The
argument can be <code>"eta"</code> (partial eta-squared) or <code>"omega"</code> (partial
omega-squared).</p></dd>

    <dt><code>bf.prior</code></dt>
<dd><p>A number between <code>0.5</code> and <code>2</code> (default <code>0.707</code>), the prior
width to use in calculating Bayes factors and posterior estimates. In
addition to numeric arguments, several named values are also recognized:
<code>"medium"</code>, <code>"wide"</code>, and <code>"ultrawide"</code>, corresponding to <em>r</em> scale values
of <code>1/2</code>, <code>sqrt(2)/2</code>, and <code>1</code>, respectively. In case of an ANOVA, this
value corresponds to scale for fixed effects.</p></dd>

    <dt><code>bf.message</code></dt>
<dd><p>Logical that decides whether to display Bayes Factor in
favor of the <em>null</em> hypothesis. This argument is relevant only <strong>for
parametric test</strong> (Default: <code>TRUE</code>).</p></dd>

    <dt><code>results.subtitle</code></dt>
<dd><p>Decides whether the results of statistical tests are
to be displayed as a subtitle (Default: <code>TRUE</code>). If set to <code>FALSE</code>, only
the plot will be returned.</p></dd>

    <dt><code>xlab</code></dt>
<dd><p>Label for <code>x</code> axis variable. If <code>NULL</code> (default),
variable name for <code>x</code> will be used.</p></dd>

    <dt><code>ylab</code></dt>
<dd><p>Labels for <code>y</code> axis variable. If <code>NULL</code> (default),
variable name for <code>y</code> will be used.</p></dd>

    <dt><code>caption</code></dt>
<dd><p>The text for the plot caption. This argument is relevant only
if <code>bf.message = FALSE</code>.</p></dd>

    <dt><code>title</code></dt>
<dd><p>The text for the plot title.</p></dd>

    <dt><code>subtitle</code></dt>
<dd><p>The text for the plot subtitle. Will work only if
<code>results.subtitle = FALSE</code>.</p></dd>

    <dt><code>digits</code></dt>
<dd><p>Number of digits for rounding or significant figures. May also
be <code>"signif"</code> to return significant figures or <code>"scientific"</code>
to return scientific notation. Control the number of digits by adding the
value as suffix, e.g. <code>digits = "scientific4"</code> to have scientific
notation with 4 decimal places, or <code>digits = "signif5"</code> for 5
significant figures (see also <code><a href="https://rdrr.io/r/base/Round.html" class="external-link">signif()</a></code>).</p></dd>

    <dt><code>var.equal</code></dt>
<dd><p>a logical variable indicating whether to treat the
    two variances as being equal. If <code>TRUE</code> then the pooled
    variance is used to estimate the variance otherwise the Welch
    (or Satterthwaite) approximation to the degrees of freedom is used.</p></dd>

    <dt><code>conf.level</code></dt>
<dd><p>Scalar between <code>0</code> and <code>1</code> (default: <code>95%</code>
confidence/credible intervals, <code>0.95</code>). If <code>NULL</code>, no confidence intervals
will be computed.</p></dd>

    <dt><code>nboot</code></dt>
<dd><p>Number of bootstrap samples for computing confidence interval
for the effect size (Default: <code>100L</code>).</p></dd>

    <dt><code>tr</code></dt>
<dd><p>Trim level for the mean when carrying out <code>robust</code> tests. In case
of an error, try reducing the value of <code>tr</code>, which is by default set to
<code>0.2</code>. Lowering the value might help.</p></dd>

    <dt><code>centrality.plotting</code></dt>
<dd><p>Logical that decides whether centrality tendency
measure is to be displayed as a point with a label (Default: <code>TRUE</code>).
Function decides which central tendency measure to show depending on the
<code>type</code> argument.</p><ul><li><p><strong>mean</strong> for parametric statistics</p></li>
<li><p><strong>median</strong> for non-parametric statistics</p></li>
<li><p><strong>trimmed mean</strong> for robust statistics</p></li>
<li><p><strong>MAP estimator</strong> for Bayesian statistics</p></li>
</ul><p>If you want default centrality parameter, you can specify this using
<code>centrality.type</code> argument.</p></dd>

    <dt><code>centrality.type</code></dt>
<dd><p>Decides which centrality parameter is to be displayed.
The default is to choose the same as <code>type</code> argument. You can specify this
to be:</p><ul><li><p><code>"parameteric"</code> (for <strong>mean</strong>)</p></li>
<li><p><code>"nonparametric"</code> (for <strong>median</strong>)</p></li>
<li><p><code>robust</code> (for <strong>trimmed mean</strong>)</p></li>
<li><p><code>bayes</code> (for <strong>MAP estimator</strong>)</p></li>
</ul><p>Just as <code>type</code> argument, abbreviations are also accepted.</p></dd>

    <dt><code>centrality.point.args,centrality.label.args</code></dt>
<dd><p>A list of additional aesthetic
arguments to be passed to <code><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">ggplot2::geom_point()</a></code> and
<code><a href="https://ggrepel.slowkow.com/reference/geom_text_repel.html" class="external-link">ggrepel::geom_label_repel()</a></code> geoms, which are involved in mean plotting.</p></dd>

    <dt><code>point.args</code></dt>
<dd><p>A list of additional aesthetic arguments to be passed to
the <code><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">ggplot2::geom_point()</a></code>.</p></dd>

    <dt><code>boxplot.args</code></dt>
<dd><p>A list of additional aesthetic arguments passed on to
<code><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">ggplot2::geom_boxplot()</a></code>.</p></dd>

    <dt><code>violin.args</code></dt>
<dd><p>A list of additional aesthetic arguments to be passed to
the <code><a href="https://ggplot2.tidyverse.org/reference/geom_violin.html" class="external-link">ggplot2::geom_violin()</a></code>.</p></dd>

    <dt><code>ggsignif.args</code></dt>
<dd><p>A list of additional aesthetic
arguments to be passed to <code><a href="https://const-ae.github.io/ggsignif/reference/stat_signif.html" class="external-link">ggsignif::geom_signif()</a></code>.</p></dd>

    <dt><code>ggtheme</code></dt>
<dd><p>A <code>{ggplot2}</code> theme. Default value is
<code><a href="https://indrajeetpatil.github.io/ggstatsplot/reference/theme_ggstatsplot.html" class="external-link">theme_ggstatsplot()</a></code>. Any of the <code>{ggplot2}</code> themes (e.g.,
<code><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">ggplot2::theme_bw()</a></code>), or themes from extension packages are allowed
(e.g., <code>ggthemes::theme_fivethirtyeight()</code>, <code>hrbrthemes::theme_ipsum_ps()</code>,
etc.). But note that sometimes these themes will remove some of the details
that <code>{ggstatsplot}</code> plots typically contains. For example, if relevant,
<code><a href="https://indrajeetpatil.github.io/ggstatsplot/reference/ggbetweenstats.html" class="external-link">ggbetweenstats()</a></code> shows details about multiple comparison test as a
label on the secondary Y-axis. Some themes (e.g.
<code>ggthemes::theme_fivethirtyeight()</code>) will remove the secondary Y-axis and
thus the details as well.</p></dd>

    <dt><code>package,palette</code></dt>
<dd><p>Name of the package from which the given palette is to
be extracted. The available palettes and packages can be checked by running
<code>View(paletteer::palettes_d_names)</code>.</p></dd>

    <dt><code>ggplot.component</code></dt>
<dd><p>A <code>ggplot</code> component to be added to the plot prepared
by <code>{ggstatsplot}</code>. This argument is primarily helpful for <code>grouped_</code>
variants of all primary functions. Default is <code>NULL</code>. The argument should
be entered as a <code>{ggplot2}</code> function or a list of <code>{ggplot2}</code> functions.</p></dd>


</dl></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>R object.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="load_example_results.html">load_example_results</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">## Reduce the CTD list for example purposes</span></span></span>
<span class="r-in"><span><span class="va">ctd_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="load_example_CTD.html">load_example_ctd</a></span><span class="op">(</span><span class="st">"ctd_DescartesHuman.rds"</span>,</span></span>
<span class="r-in"><span>                             multi_dataset <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading ctd_DescartesHuman.rds</span>
<span class="r-in"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="add_symptom_results.html">add_symptom_results</a></span><span class="op">(</span>results<span class="op">=</span><span class="va">results</span><span class="op">[</span><span class="va">ctd</span><span class="op">==</span><span class="st">"DescartesHuman"</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>                               ctd_list<span class="op">=</span><span class="va">ctd_list</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding symptom-level results.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2025-05-06</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Subsetting results by q_threshold and effect.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 19,929 associations remain after filtering.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding genes and disease IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2025-05-06</span>
<span class="r-in"><span><span class="co">## Reduce the number of diseases for example purposes</span></span></span>
<span class="r-in"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">[</span><span class="va">disease_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">disease_id</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="co">#### Multiple phenotypes per disease #####</span></span></span>
<span class="r-in"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/add_.html" class="external-link">add_gpt_annotations</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating ontology terms to ids.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2025-05-06</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 151 phenotypes do not have matching HPO IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading in GPT annotations for 16,982 phenotypes.</span>
<span class="r-in"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">plot_differential_outcomes</span><span class="op">(</span><span class="va">results</span>,</span></span>
<span class="r-in"><span>                           facet_var <span class="op">=</span> <span class="st">"disease_name"</span>,</span></span>
<span class="r-in"><span>                           y_var <span class="op">=</span> <span class="st">"severity_score_gpt"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding disease_name and disease_description.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: tidytext</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Robert Gordon-Smith, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

