<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Prioritise targets grid — prioritise_targets_grid • MSTExplorer</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Prioritise targets grid — prioritise_targets_grid"><meta property="og:description" content="Plot the output of prioritise_targets
as a grid."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MSTExplorer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MSTExplorer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/MSTExplorer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Prioritise targets grid</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MSTExplorer/blob/dev-hiru/R/prioritise_targets_grid.R" class="external-link"><code>R/prioritise_targets_grid.R</code></a></small>
    <div class="hidden name"><code>prioritise_targets_grid.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Plot the output of <a href="prioritise_targets.html">prioritise_targets</a>
as a grid.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">prioritise_targets_grid</span><span class="op">(</span></span>
<span>  <span class="va">top_targets</span>,</span>
<span>  res_class <span class="op">=</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/gpt_annot_class.html" class="external-link">gpt_annot_class</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  n_per_class <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  keep_severity_class <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"profound"</span>, <span class="st">"severe"</span><span class="op">)</span>,</span>
<span>  keep_physical_malformations <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  species_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Homo sapiens"</span>, <span class="st">"Macaca mulatta"</span>, <span class="st">"Mus musculus"</span>, <span class="st">"Danio rerio"</span>,</span>
<span>    <span class="st">"Drosophila melanogaster"</span>, <span class="st">"Caenorhabditis elegans"</span><span class="op">)</span>,</span>
<span>  legend.position <span class="op">=</span> <span class="st">"left"</span>,</span>
<span>  keep_ont_levels <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  width <span class="op">=</span> <span class="fl">70</span>,</span>
<span>  widths1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">4</span><span class="op">)</span>,</span>
<span>  widths2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0.8</span><span class="op">)</span>,</span>
<span>  show_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-top-targets">top_targets<a class="anchor" aria-label="anchor" href="#arg-top-targets"></a></dt>
<dd><p>output of <a href="prioritise_targets.html">prioritise_targets</a>.</p></dd>


<dt id="arg-res-class">res_class<a class="anchor" aria-label="anchor" href="#arg-res-class"></a></dt>
<dd><p>Output of the <a href="https://rdrr.io/pkg/HPOExplorer/man/gpt_annot_class.html" class="external-link">gpt_annot_class</a> function.</p></dd>


<dt id="arg-n-per-class">n_per_class<a class="anchor" aria-label="anchor" href="#arg-n-per-class"></a></dt>
<dd><p>Number of phenotypes per severity class to include.</p></dd>


<dt id="arg-keep-severity-class">keep_severity_class<a class="anchor" aria-label="anchor" href="#arg-keep-severity-class"></a></dt>
<dd><p>Phenotypes to keep based on severity classes.</p></dd>


<dt id="arg-keep-physical-malformations">keep_physical_malformations<a class="anchor" aria-label="anchor" href="#arg-keep-physical-malformations"></a></dt>
<dd><p>Phenotypes to keep based on physical
malformation frequency (0=never, 1=rarely, 2=often, 3=always).</p></dd>


<dt id="arg-species-list">species_list<a class="anchor" aria-label="anchor" href="#arg-species-list"></a></dt>
<dd><p>Species to include in orthologous genes grid.</p></dd>


<dt id="arg-legend-position">legend.position<a class="anchor" aria-label="anchor" href="#arg-legend-position"></a></dt>
<dd><p>the default position of legends ("none", "left",
"right", "bottom", "top", "inside")</p></dd>


<dt id="arg-keep-ont-levels">keep_ont_levels<a class="anchor" aria-label="anchor" href="#arg-keep-ont-levels"></a></dt>
<dd><p>Only keep phenotypes at certain <em>absolute</em>
ontology levels to keep.
See <a href="https://rdrr.io/pkg/HPOExplorer/man/add_.html" class="external-link">add_ont_lvl</a> for details.</p></dd>


<dt id="arg-width">width<a class="anchor" aria-label="anchor" href="#arg-width"></a></dt>
<dd><p>Width of the saved plot.</p></dd>


<dt id="arg-widths-">widths1<a class="anchor" aria-label="anchor" href="#arg-widths-"></a></dt>
<dd><p>Proportional widths of severity annotation heatmap
(left subplot) and the top targets/orthologous genes grid (right subplot).</p></dd>


<dt id="arg-widths-">widths2<a class="anchor" aria-label="anchor" href="#arg-widths-"></a></dt>
<dd><p>Proportional widths of the top targets/orthologous genes grid.</p></dd>


<dt id="arg-show-plot">show_plot<a class="anchor" aria-label="anchor" href="#arg-show-plot"></a></dt>
<dd><p>Print the plot to the console.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Arguments passed on to <code><a href="https://rdrr.io/pkg/HPOExplorer/man/plot_top_phenos.html" class="external-link">HPOExplorer::plot_top_phenos</a></code></p><dl><dt><code>keep_descendants</code></dt>
<dd><p>Terms whose descendants should be kept
(including themselves).
 Set to <code>NULL</code> (default) to skip this filtering step.</p></dd>

    <dt><code>annotation_order</code></dt>
<dd><p>The order of the annotations to include.</p></dd>

    <dt><code>split_by_congenital</code></dt>
<dd><p>Split the phenotypes by congenital onset
(congenital = always/often, noncongenital = never/rarely).</p></dd>

    <dt><code>axis.text.x</code></dt>
<dd><p>Whether to include x-axis text in top and bottom subplots.</p></dd>


</dl></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A named list with data and patchwork object.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">top_targets</span> <span class="op">&lt;-</span> <span class="fu">MSTExplorer</span><span class="fu">::</span><span class="va"><a href="example_targets.html">example_targets</a></span><span class="op">$</span><span class="va">top_targets</span></span></span>
<span class="r-in"><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">prioritise_targets_grid</span><span class="op">(</span>top_targets <span class="op">=</span> <span class="va">top_targets</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping cell types to cell ontology terms.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding stage information.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding disease_name and disease_description.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating ontology terms to ids.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2024-12-12</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 383 phenotypes do not have matching HPO IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading in GPT annotations for 16,753 phenotypes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Assigning severity classes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding level-2 ancestor to each HPO ID.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ancestor metadata.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Ancestor metadata already present. Use force_new=TRUE to overwrite.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating ontology terms to ids.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Keeping descendants of 1 term(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 18,379 terms remain after filtering.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1,150 associations remain after filtering.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting absolute ontology level for 19,025 IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding disease_name and disease_description.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.table format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting human ==&gt; Macaca mulatta orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Macaca mulatta</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9544</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in Macaca mulatta.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 7 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 7 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    -10 / 7 (-143%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    17 / 7 (243%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.table format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting human ==&gt; Mus musculus orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Mus musculus</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in Mus musculus.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 7 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 7 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    -10 / 7 (-143%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    17 / 7 (243%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.table format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting human ==&gt; Danio rerio orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Danio rerio</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7955</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in Danio rerio.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 7 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 7 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    -10 / 7 (-143%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    17 / 7 (243%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.table format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting human ==&gt; Drosophila melanogaster orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Drosophila melanogaster</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7227</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in Drosophila melanogaster.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    0 / 7 (0%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    7 / 7 (100%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.table format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting human ==&gt; Caenorhabditis elegans orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Caenorhabditis elegans</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 6239</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in Caenorhabditis elegans.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 5 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 5 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 1 genes that have multiple input_gene per ortholog_gene (many:1).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    -4 / 7 (-57%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    11 / 7 (157%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding disease_name and disease_description.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.table format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 13 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting human ==&gt; Macaca mulatta orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Macaca mulatta</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9544</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in Macaca mulatta.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 6 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 6 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    -4 / 7 (-57%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    11 / 7 (157%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.table format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 13 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting human ==&gt; Mus musculus orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Mus musculus</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in Mus musculus.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 7 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 7 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    -6 / 7 (-86%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    13 / 7 (186%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.table format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 13 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting human ==&gt; Danio rerio orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Danio rerio</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7955</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in Danio rerio.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 7 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 7 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    -6 / 7 (-86%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    13 / 7 (186%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.table format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 13 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting human ==&gt; Drosophila melanogaster orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Drosophila melanogaster</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7227</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in Drosophila melanogaster.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    0 / 7 (0%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    7 / 7 (100%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.table format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 13 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting human ==&gt; Caenorhabditis elegans orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Caenorhabditis elegans</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 6239</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in Caenorhabditis elegans.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    -3 / 7 (-43%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    10 / 7 (143%)</span>
<span class="r-plt img"><img src="prioritise_targets_grid-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Robert Gordon-Smith, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

