<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Prioritise target genes — prioritise_targets • MultiEWCE</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Prioritise target genes — prioritise_targets"><meta property="og:description" content="Prioritise target genes based on a procedure:
Disease-level: keep_deaths: 
Keep only diseases with a certain age of death.
Phenotype-level: remove_descendants: 
Remove phenotypes belonging to a certain branch of the HPO,
as defined by an ancestor term.
Phenotype-level: keep_ont_levels: 
Keep only phenotypes at certain absolute ontology levels within the HPO.
Phenotype-level: pheno_ndiseases_threshold: 
The maximum number of diseases each phenotype can be associated with.
Phenotype-level: keep_tiers: 
Keep only phenotypes with high severity Tiers.
Phenotype-level: severity_threshold: 
Keep only phenotypes with mean Severity equal to or below the threshold.
Symptom-level: pheno_frequency_threshold: 
Keep only phenotypes with mean frequency equal to or above the threshold
(i.e. how frequently a phenotype is associated with any diseases in
which it occurs).
Symptom-level: keep_onsets: 
Keep only symptoms with a certain age of onset.
Symptom-level: symptom_p_threshold: 
Uncorrected p-value threshold to filter cell type-symptom associations by.
Symptom-level: symptom_intersection_size_threshold: 
Minimum number of genes overlapping between a symptom gene list
(phenotype-associated genes in the context of a particular disease)
and the celltype (genes in top 10/40 specificity quantiles).
Cell type-level: q_threshold: 
Keep only cell type-phenotype association results at q&amp;lt;=0.05.
Cell type-level: fold_threshold: 
Keep only cell type-phenotype association results at fold_change&amp;gt;=1.
Cell type-level: keep_celltypes: 
Keep only terminally differentiated cell types.
Gene-level: keep_seqnames: 
Remove genes on non-standard chromosomes.
Gene-level: gene_size: 
Keep only genes &amp;lt;4.3kb in length.
Gene-level: keep_biotypes: 
Keep only genes belonging to certain biotypes.
Gene-level: gene_frequency_threshold: 
Keep only genes at or above a certain mean frequency threshold
(i.e. how frequently a gene is associated with a given phenotype
when observed within a disease).
Gene-level: keep_specificity_quantiles: 
Keep only genes in top specificity quantiles
from the cell type dataset (ctd).
Gene-level: keep_mean_exp_quantiles: 
Keep only genes in top mean expression quantiles
from the cell type dataset (ctd).
Gene-level: symptom_gene_overlap: 
Ensure that genes nominated at the phenotype-level also
appear in the genes overlapping at the cell type-specific symptom-level.
All levels: top_n: 
Sort candidate targets by a preferred order of metrics and
only return the top N targets per cell type-phenotype combination.

"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MultiEWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MultiEWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/MultiEWCE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Prioritise target genes</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MultiEWCE/blob/HEAD/R/prioritise_targets.R" class="external-link"><code>R/prioritise_targets.R</code></a></small>
    <div class="hidden name"><code>prioritise_targets.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Prioritise target genes based on a procedure:<br></p><ol><li><p>Disease-level: <code>keep_deaths</code>: 
Keep only diseases with a certain age of death.</p></li>
<li><p>Phenotype-level: <code>remove_descendants</code>: 
Remove phenotypes belonging to a certain branch of the HPO,
as defined by an ancestor term.</p></li>
<li><p>Phenotype-level: <code>keep_ont_levels</code>: 
Keep only phenotypes at certain absolute ontology levels within the HPO.</p></li>
<li><p>Phenotype-level: <code>pheno_ndiseases_threshold</code>: 
The maximum number of diseases each phenotype can be associated with.</p></li>
<li><p>Phenotype-level: <code>keep_tiers</code>: 
Keep only phenotypes with high severity Tiers.</p></li>
<li><p>Phenotype-level: <code>severity_threshold</code>: 
Keep only phenotypes with mean Severity equal to or below the threshold.</p></li>
<li><p>Symptom-level: <code>pheno_frequency_threshold</code>: 
Keep only phenotypes with mean frequency equal to or above the threshold
(i.e. how frequently a phenotype is associated with any diseases in
which it occurs).</p></li>
<li><p>Symptom-level: <code>keep_onsets</code>: 
Keep only symptoms with a certain age of onset.</p></li>
<li><p>Symptom-level: <code>symptom_p_threshold</code>: 
Uncorrected p-value threshold to filter cell type-symptom associations by.</p></li>
<li><p>Symptom-level: <code>symptom_intersection_size_threshold</code>: 
Minimum number of genes overlapping between a symptom gene list
(phenotype-associated genes in the context of a particular disease)
and the celltype (genes in top 10/40 specificity quantiles).</p></li>
<li><p>Cell type-level: <code>q_threshold</code>: 
Keep only cell type-phenotype association results at q&lt;=0.05.</p></li>
<li><p>Cell type-level: <code>fold_threshold</code>: 
Keep only cell type-phenotype association results at fold_change&gt;=1.</p></li>
<li><p>Cell type-level: <code>keep_celltypes</code>: 
Keep only terminally differentiated cell types.</p></li>
<li><p>Gene-level: <code>keep_seqnames</code>: 
Remove genes on non-standard chromosomes.</p></li>
<li><p>Gene-level: <code>gene_size</code>: 
Keep only genes &lt;4.3kb in length.</p></li>
<li><p>Gene-level: <code>keep_biotypes</code>: 
Keep only genes belonging to certain biotypes.</p></li>
<li><p>Gene-level: <code>gene_frequency_threshold</code>: 
Keep only genes at or above a certain mean frequency threshold
(i.e. how frequently a gene is associated with a given phenotype
when observed within a disease).</p></li>
<li><p>Gene-level: <code>keep_specificity_quantiles</code>: 
Keep only genes in top specificity quantiles
from the cell type dataset (<code>ctd</code>).</p></li>
<li><p>Gene-level: <code>keep_mean_exp_quantiles</code>: 
Keep only genes in top mean expression quantiles
from the cell type dataset (<code>ctd</code>).</p></li>
<li><p>Gene-level: <code>symptom_gene_overlap</code>: 
Ensure that genes nominated at the phenotype-level also
appear in the genes overlapping at the cell type-specific symptom-level.</p></li>
<li><p>All levels: <code>top_n</code>: 
Sort candidate targets by a preferred order of metrics and
only return the top N targets per cell type-phenotype combination.</p></li>
</ol></div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">prioritise_targets</span><span class="op">(</span></span>
<span>  results <span class="op">=</span> <span class="fu"><a href="load_example_results.html">load_example_results</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  ctd <span class="op">=</span> <span class="fu"><a href="load_example_CTD.html">load_example_ctd</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  annotLevel <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  phenotype_to_genes <span class="op">=</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/load_phenotype_to_genes.html" class="external-link">load_phenotype_to_genes</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  hpo <span class="op">=</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/get_hpo.html" class="external-link">get_hpo</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  keep_deaths <span class="op">=</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/list_deaths.html" class="external-link">list_deaths</a></span><span class="op">(</span>exclude <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Miscarriage"</span>, <span class="st">"Stillbirth"</span>,</span>
<span>    <span class="st">"Prenatal death"</span><span class="op">)</span>, include_na <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  remove_descendants <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Clinical course"</span><span class="op">)</span>,</span>
<span>  keep_ont_levels <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pheno_ndiseases_threshold <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  keep_tiers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="cn">NA</span><span class="op">)</span>,</span>
<span>  severity_threshold_max <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  severity_threshold <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="cn">NA</span><span class="op">)</span>,</span>
<span>  pheno_frequency_threshold <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  keep_onsets <span class="op">=</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/list_onsets.html" class="external-link">list_onsets</a></span><span class="op">(</span>exclude <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Antenatal"</span>, <span class="st">"Fetal"</span>, <span class="st">"Congenital"</span><span class="op">)</span>,</span>
<span>    include_na <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  q_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  fold_threshold <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  symptom_p_threshold <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  symptom_intersection_size_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  keep_celltypes <span class="op">=</span> <span class="fu"><a href="terminal_celltypes.html">terminal_celltypes</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">CellType</span>,</span>
<span>  keep_evidence <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">6</span><span class="op">)</span>,</span>
<span>  keep_seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">22</span><span class="op">)</span>, <span class="st">"X"</span>, <span class="st">"Y"</span><span class="op">)</span>,</span>
<span>  gene_size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>min <span class="op">=</span> <span class="fl">0</span>, max <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span>,</span>
<span>  gene_frequency_threshold <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  keep_biotypes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  keep_specificity_quantiles <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  keep_mean_exp_quantiles <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">40</span><span class="op">)</span>,</span>
<span>  symptom_gene_overlap <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sort_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>tier_merge <span class="op">=</span> <span class="fl">1</span>, Severity_score_mean <span class="op">=</span> <span class="fl">1</span>, q <span class="op">=</span> <span class="fl">1</span>, fold_change <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>,</span>
<span>    specificity <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, mean_exp <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, pheno_freq_mean <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, gene_freq_mean <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, width <span class="op">=</span></span>
<span>    <span class="fl">1</span><span class="op">)</span>,</span>
<span>  top_n <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"disease_id"</span>, <span class="st">"hpo_id"</span>, <span class="st">"CellType"</span><span class="op">)</span>,</span>
<span>  return_report <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>results</dt>
<dd><p>The cell type-phenotype enrichment results generated by
<a href="gen_results.html">gen_results</a>
and merged together with <a href="merge_results.html">merge_results</a>.</p></dd>


<dt>ctd</dt>
<dd><p>Cell Type Data List generated using
<a href="https://rdrr.io/pkg/EWCE/man/generate_celltype_data.html" class="external-link">generate_celltype_data</a>.</p></dd>


<dt>annotLevel</dt>
<dd><p>An integer indicating which level of <code>sct_data</code> to
analyse (<em>Default: 1</em>).</p></dd>


<dt>phenotype_to_genes</dt>
<dd><p>Output of
<a href="https://rdrr.io/pkg/HPOExplorer/man/load_phenotype_to_genes.html" class="external-link">load_phenotype_to_genes</a> mapping phenotypes
to gene annotations.</p></dd>


<dt>hpo</dt>
<dd><p>Human Phenotype Ontology object, loaded from <span class="pkg">ontologyIndex</span>.</p></dd>


<dt>keep_deaths</dt>
<dd><p>The age of death associated with each HPO ID to keep.
If &gt;1 age of death is associated with the term,
only the earliest age is considered.
See <a href="https://rdrr.io/pkg/HPOExplorer/man/add_death.html" class="external-link">add_death</a> for details.</p></dd>


<dt>remove_descendants</dt>
<dd><p>Remove HPO terms that are descendants of a given
ancestral HPO term. Ancestral terms be provided as a character vector of
phenotype names (e.g. <code>c("Clinical course")</code>),
HPO IDs (e.g. <code>"HP:0031797" </code>) or a mixture of the two.
See <a href="https://rdrr.io/pkg/HPOExplorer/man/add_ancestor.html" class="external-link">add_ancestor</a> for details.</p></dd>


<dt>keep_ont_levels</dt>
<dd><p>Only keep phenotypes at certain <em>absolute</em>
ontology levels to keep.
See <a href="https://rdrr.io/pkg/HPOExplorer/man/add_ont_lvl.html" class="external-link">add_ont_lvl</a> for details.</p></dd>


<dt>pheno_ndiseases_threshold</dt>
<dd><p>Filter phenotypes by the
maximum number of diseases they are associated with.</p></dd>


<dt>keep_tiers</dt>
<dd><p>Tiers from <a href="https://rdrr.io/pkg/HPOExplorer/man/hpo_tiers.html" class="external-link">hpo_tiers</a> to keep.
Include <code>NA</code> if you wish to retain phenotypes that
do not have any Tier assignment.</p></dd>


<dt>severity_threshold_max</dt>
<dd><p>The max severity score that a phenotype can
have across any disease.</p></dd>


<dt>severity_threshold</dt>
<dd><p>Only keep phenotypes with a mean
severity score (averaged across multiple associated diseases) below the
set threshold. The severity score ranges from 1-4 where 1 is the MOST severe.
Include <code>NA</code> if you wish to retain phenotypes that
do not have any severity score.</p></dd>


<dt>pheno_frequency_threshold</dt>
<dd><p>Only keep phenotypes with frequency
above the set threshold. Frequency ranges from 0-100 where 100 is
a phenotype that occurs 100% of the time in all associated diseases.
Include <code>NA</code> if you wish to retain phenotypes that
do not have any frequency data.
See <a href="https://rdrr.io/pkg/HPOExplorer/man/add_pheno_frequency.html" class="external-link">add_pheno_frequency</a> for details.</p></dd>


<dt>keep_onsets</dt>
<dd><p>The age of onset associated with each HPO ID to keep.
If &gt;1 age of onset is associated with the term,
only the earliest age is considered.
See <a href="https://rdrr.io/pkg/HPOExplorer/man/add_onset.html" class="external-link">add_onset</a> for details.</p></dd>


<dt>q_threshold</dt>
<dd><p>The q value threshold to subset the <code>results</code> by.</p></dd>


<dt>fold_threshold</dt>
<dd><p>The minimum fold change in specific expression
to subset the <code>results</code> by.</p></dd>


<dt>symptom_p_threshold</dt>
<dd><p>The p-value threshold of celltype-symptom
enrichment results (using  <a href="gen_overlap.html">gen_overlap</a>).
Here, "symptoms" are defined as the presentation of a phenotype in the
context of a particular disease.
In other words:
phenotype (hpo_id) + disease (disease_id) = symptom (hpo_id.disease_id)</p></dd>


<dt>symptom_intersection_size_threshold</dt>
<dd><p>The minimum number of intersecting genes between a symptom and a celltype to
consider it a significant enrichment. Refers to the result from
<a href="gen_overlap.html">gen_overlap</a>.</p></dd>


<dt>keep_celltypes</dt>
<dd><p>Cell type to keep.</p></dd>


<dt>keep_evidence</dt>
<dd><p>The evidence scores of each gene-disease
association to keep.</p></dd>


<dt>keep_seqnames</dt>
<dd><p>Chromosomes to keep.</p></dd>


<dt>gene_size</dt>
<dd><p>Min/max gene size (important for therapeutics design).</p></dd>


<dt>gene_frequency_threshold</dt>
<dd><p>Only keep genes with frequency
above the set threshold. Frequency ranges from 0-100 where 100 is
a gene that occurs 100% of the time in a given phenotype.
Include <code>NA</code> if you wish to retain genes that
do not have any frequency data.
See <a href="https://rdrr.io/pkg/HPOExplorer/man/add_gene_frequency.html" class="external-link">add_gene_frequency</a> for details.</p></dd>


<dt>keep_biotypes</dt>
<dd><p>Which gene biotypes to keep.
(e.g. "protein_coding", "processed_transcript", "snRNA",
"lincRNA", "snoRNA", "IG_C_gene")</p></dd>


<dt>keep_specificity_quantiles</dt>
<dd><p>Which cell type
specificity quantiles to keep (max quantile is 40).</p></dd>


<dt>keep_mean_exp_quantiles</dt>
<dd><p>Which cell type
mean expression quantiles to keep (max quantile is 40).</p></dd>


<dt>symptom_gene_overlap</dt>
<dd><p>The gene for a particular symptom
(phenotype + disease) must appear in the celltype-symptom
enrichment results.</p></dd>


<dt>sort_cols</dt>
<dd><p>How to sort the rows using <a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html" class="external-link">setorderv</a>.
<code>names(sort_cols)</code> will be supplied to the <code>cols=</code> argument
and values will be supplied to the <code>order=</code> argument.</p></dd>


<dt>top_n</dt>
<dd><p>Top N genes to keep when grouping by <code>group_vars</code>.</p></dd>


<dt>group_vars</dt>
<dd><p>Columns to group by when selecting <code>top_n</code> genes.</p></dd>


<dt>return_report</dt>
<dd><p>If <code>TRUE</code>, will return a named list containing a
<code>report</code> that shows the number of
phenotypes/celltypes/genes remaining after each filtering step.</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A data.table of the prioritised phenotype- and
celltype-specific gene targets.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Term key:<br></p><ul><li><p>Disease: A disease defined in the database
OMIM, DECIPHER and/or Orphanet.</p></li>
<li><p>Phenotype: A clinical feature associated with one or more diseases.</p></li>
<li><p>Symptom: A phenotype within the context of a particular disease.
Within a given phenotype, there may be multiple symptoms with
partially overlapping genetic mechanisms.</p></li>
<li><p>Assocation: A cell type-specific enrichment test result conducted
at the disease-level, phenotype-level, or symptom-level.</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">results</span> <span class="op">=</span> <span class="fu"><a href="load_example_results.html">load_example_results</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">5000</span><span class="op">)</span>,<span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">prioritise_targets</span><span class="op">(</span>results<span class="op">=</span><span class="va">results</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritising gene targets.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding term definitions.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> All local files already up-to-date!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating phenos with Disease</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype.hpoa</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2023-10-09</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding disease metadata: Definitions, Preferred.Label</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Orphanet metadata.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing OMIM metadata.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1178 / 2469 (47.71%) disease_name missing.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1867 / 2335 (79.96%) Definitions missing.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating phenos with MONDO metadata.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> All local files already up-to-date!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 7 / 2335 (0.3%) MONDO_ID missing.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2037 / 2335 (87.24%) MONDO_name missing.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2289 / 2335 (98.03%) MONDO_definition missing.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1828 / 2335 (78.29%) Definitions missing.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='start' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 5,000 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 3 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 2,335 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 14</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering @ q-value &lt;= 0.05</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='q_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 5,000 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 3 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 2,335 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 14</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering @ fold-change &gt;= 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='fold_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 20 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 20 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='symptom_p_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 20 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 20 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='symptom_intersection_size_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 20 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 20 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating phenos with AgeOfDeath.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating all phenotypes to HPO IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> All local files already up-to-date!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Returning a dictionary of phenotypes (different order as input).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_deaths' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding level-3 ancestor to each HPO ID.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> All local files already up-to-date!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing remove descendants of: 'Clinical course'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating all phenotypes to HPO IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Returning a dictionary of phenotypes (different order as input).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='remove_descendants' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting absolute ontology level for 1 HPO IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> All local files already up-to-date!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_ont_levels' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating phenos with onset.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating all phenotypes to HPO IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> All local files already up-to-date!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Returning a dictionary of phenotypes (different order as input).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_onsets' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating phenos with Tiers.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_tiers' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating phenos with modifiers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='severity_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='severity_threshold_max' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating phenos with n_diseases</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2023-10-09</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: genes_to_phenotype.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2023-10-09</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype.hpoa</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2023-10-09</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='pheno_ndiseases_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating phenotype frequencies.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='pheno_frequency_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 / 1 of cell types kept.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_celltypes' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting phenos to GRanges.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2023-10-09</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: ensembldb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering metadata for 2 unique genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: EnsDb.Hsapiens.v75</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='symptom_gene_overlap' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 2 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering by keep_seqnames.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_seqnames' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 2 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering by gene-disease association evidence.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating gene-disease associations with Evidence score</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering data from GenCC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing cached file.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: 2023-11-07</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_evidence' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 2 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering by gene size.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2 / 2 genes kept.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='gene_size' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 2 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_biotypes' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 2 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2023-10-09</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_specificity_quantiles' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 2 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_mean_exp_quantiles' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 2 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating gene frequencies.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: genes_to_phenotype.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2023-10-09</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='gene_frequency_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 2 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sorting rows.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='end' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 2 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 2</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robert Gordon-Smith, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

