<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Prioritise target genes — prioritise_targets • MSTExplorer</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Prioritise target genes — prioritise_targets"><meta property="og:description" content='Prioritise target genes based on a procedure:
Disease-level: keep_deaths:
Keep only diseases with a certain age of death.
Disease-level: severity_threshold_max:

Keep only diseases annotated as a certain degree of severity or greater
 (filters on maximum severity per disease).
Phenotype-level: prune_ancestors:

Remove redundant ancestral phenotypes when at least one of their
 descendants already exist.
Phenotype-level: keep_descendants:

Remove phenotypes belonging to a certain branch of the HPO,
 as defined by an ancestor term.
Phenotype-level: keep_ont_levels:
Keep only phenotypes at certain absolute ontology levels within the HPO.
Phenotype-level: pheno_ndiseases_threshold:
The maximum number of diseases each phenotype can be associated with.
Phenotype-level: keep_tiers:
Keep only phenotypes with high severity Tiers.
Phenotype-level: severity_threshold:
Keep only phenotypes with mean Severity equal to or below the threshold.
Phenotype-level: gpt_filters:

Keep only phenotypes with certain GPT annotations in specific
 severity metrics.
Phenotype-level: severity_score_gpt_threshold:
Keep only phenotypes with a minimum GPT severity score.
Phenotype-level: info_content_threshold:

Keep only phenotypes with a minimum information criterion score
 (computed from the HPO).
Symptom-level: pheno_frequency_threshold:

Keep only phenotypes with mean frequency equal to or above the threshold
 (i.e. how frequently a phenotype is associated with any diseases in
 which it occurs).
Symptom-level: keep_onsets:
Keep only symptoms with a certain age of onset.
Symptom-level: symptom_p_threshold:
Uncorrected p-value threshold to filter cell type-symptom associations by.
Symptom-level: symptom_intersection_threshold:

Minimum proportion of genes overlapping between a symptom gene list
 (phenotype-associated genes in the context of a particular disease)
 and the phenotype-cell type association driver genes.
Cell type-level: q_threshold:

Keep only cell type-phenotype association results at q&amp;lt;=0.05.
Cell type-level: effect_threshold:
Keep only cell type-phenotype association results at effect size&amp;gt;=1.
Cell type-level: keep_celltypes:
Keep only terminally differentiated cell types.
Gene-level: keep_chr:
Remove genes on non-standard chromosomes.
Gene-level: evidence_score_threshold:

Remove genes that are below an aggregate phenotype-gene
 evidence score threshold.
Gene-level: gene_size:
Keep only genes &amp;lt;4.3kb in length.
Gene-level: add_driver_genes:

Keep only genes that are driving the association with a given phenotype
 (inferred by the intersection of phenotype-associated genes and gene with
 high-specificity quantiles in the target cell type).
Gene-level: keep_biotypes:
Keep only genes belonging to certain biotypes.
Gene-level: gene_frequency_threshold:

Keep only genes at or above a certain mean frequency threshold
 (i.e. how frequently a gene is associated with a given phenotype
 when observed within a disease).
Gene-level: keep_specificity_quantiles:

Keep only genes in top specificity quantiles
 from the cell type dataset (CTD).
Gene-level: keep_mean_exp_quantiles:

Keep only genes in top mean expression quantiles
 from the cell type dataset (CTD).
Gene-level: symptom_gene_overlap:

Ensure that genes nominated at the phenotype-level also
 appear in the genes overlapping at the cell type-specific symptom-level.
All levels: sort_cols:

Sort candidate targets by one or more columns
 (e.g. "severity_score_gpt", "q").

All levels: top_n:

Only return the top N targets per variable group
 (specified with the "group_vars" argument).
 For example, setting "group_vars" to "hpo_id" and "top_n" to 1 would
 only return one target (row) per phenotype ID after sorting.

'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MSTExplorer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MSTExplorer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/MSTExplorer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Prioritise target genes</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MSTExplorer/blob/dev-hiru/R/prioritise_targets.R" class="external-link"><code>R/prioritise_targets.R</code></a></small>
    <div class="hidden name"><code>prioritise_targets.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Prioritise target genes based on a procedure:<br></p><ol><li><p>Disease-level: <code>keep_deaths</code>:
Keep only diseases with a certain age of death.</p></li>
<li><p>Disease-level: <code>severity_threshold_max</code>:
</p><div class="sourceCode"><pre><code>
Keep only diseases annotated as a certain degree of severity or greater
 (filters on maximum severity per disease).</code></pre></div></li>
<li><p>Phenotype-level: <code>prune_ancestors</code>:
</p><div class="sourceCode"><pre><code>
Remove redundant ancestral phenotypes when at least one of their
 descendants already exist.</code></pre></div></li>
<li><p>Phenotype-level: <code>keep_descendants</code>:
</p><div class="sourceCode"><pre><code>
Remove phenotypes belonging to a certain branch of the HPO,
 as defined by an ancestor term.</code></pre></div></li>
<li><p>Phenotype-level: <code>keep_ont_levels</code>:
Keep only phenotypes at certain absolute ontology levels within the HPO.</p></li>
<li><p>Phenotype-level: <code>pheno_ndiseases_threshold</code>:
The maximum number of diseases each phenotype can be associated with.</p></li>
<li><p>Phenotype-level: <code>keep_tiers</code>:
Keep only phenotypes with high severity Tiers.</p></li>
<li><p>Phenotype-level: <code>severity_threshold</code>:
Keep only phenotypes with mean Severity equal to or below the threshold.</p></li>
<li><p>Phenotype-level: <code>gpt_filters</code>:
</p><div class="sourceCode"><pre><code>
Keep only phenotypes with certain GPT annotations in specific
 severity metrics.</code></pre></div></li>
<li><p>Phenotype-level: <code>severity_score_gpt_threshold</code>:
Keep only phenotypes with a minimum GPT severity score.</p></li>
<li><p>Phenotype-level: <code>info_content_threshold</code>:
</p><div class="sourceCode"><pre><code>
Keep only phenotypes with a minimum information criterion score
 (computed from the HPO).</code></pre></div></li>
<li><p>Symptom-level: <code>pheno_frequency_threshold</code>:
</p><div class="sourceCode"><pre><code>
Keep only phenotypes with mean frequency equal to or above the threshold
 (i.e. how frequently a phenotype is associated with any diseases in
 which it occurs).</code></pre></div></li>
<li><p>Symptom-level: <code>keep_onsets</code>:
Keep only symptoms with a certain age of onset.</p></li>
<li><p>Symptom-level: <code>symptom_p_threshold</code>:
Uncorrected p-value threshold to filter cell type-symptom associations by.</p></li>
<li><p>Symptom-level: <code>symptom_intersection_threshold</code>:
</p><div class="sourceCode"><pre><code>
Minimum proportion of genes overlapping between a symptom gene list
 (phenotype-associated genes in the context of a particular disease)
 and the phenotype-cell type association driver genes.</code></pre></div></li>
<li><p>Cell type-level: <code>q_threshold</code>:
</p><div class="sourceCode"><pre><code>
Keep only cell type-phenotype association results at q&lt;=0.05.</code></pre></div></li>
<li><p>Cell type-level: <code>effect_threshold</code>:
Keep only cell type-phenotype association results at effect size&gt;=1.</p></li>
<li><p>Cell type-level: <code>keep_celltypes</code>:
Keep only terminally differentiated cell types.</p></li>
<li><p>Gene-level: <code>keep_chr</code>:
Remove genes on non-standard chromosomes.</p></li>
<li><p>Gene-level: <code>evidence_score_threshold</code>:
</p><div class="sourceCode"><pre><code>
Remove genes that are below an aggregate phenotype-gene
 evidence score threshold.</code></pre></div></li>
<li><p>Gene-level: <code>gene_size</code>:
Keep only genes &lt;4.3kb in length.</p></li>
<li><p>Gene-level: <code>add_driver_genes</code>:
</p><div class="sourceCode"><pre><code>
Keep only genes that are driving the association with a given phenotype
 (inferred by the intersection of phenotype-associated genes and gene with
 high-specificity quantiles in the target cell type).</code></pre></div></li>
<li><p>Gene-level: <code>keep_biotypes</code>:
Keep only genes belonging to certain biotypes.</p></li>
<li><p>Gene-level: <code>gene_frequency_threshold</code>:
</p><div class="sourceCode"><pre><code>
Keep only genes at or above a certain mean frequency threshold
 (i.e. how frequently a gene is associated with a given phenotype
 when observed within a disease).</code></pre></div></li>
<li><p>Gene-level: <code>keep_specificity_quantiles</code>:
</p><div class="sourceCode"><pre><code>
Keep only genes in top specificity quantiles
 from the cell type dataset (CTD).</code></pre></div></li>
<li><p>Gene-level: <code>keep_mean_exp_quantiles</code>:
</p><div class="sourceCode"><pre><code>
Keep only genes in top mean expression quantiles
 from the cell type dataset (CTD).</code></pre></div></li>
<li><p>Gene-level: <code>symptom_gene_overlap</code>:
</p><div class="sourceCode"><pre><code>
Ensure that genes nominated at the phenotype-level also
 appear in the genes overlapping at the cell type-specific symptom-level.</code></pre></div></li>
<li><p>All levels: <code>sort_cols</code>:
</p><div class="sourceCode"><pre><code>
Sort candidate targets by one or more columns
 (e.g. "severity_score_gpt", "q").
</code></pre></div></li>
<li><p>All levels: <code>top_n</code>:
</p><div class="sourceCode"><pre><code>
Only return the top N targets per variable group
 (specified with the "group_vars" argument).
 For example, setting "group_vars" to "hpo_id" and "top_n" to 1 would
 only return one target (row) per phenotype ID after sorting.</code></pre></div></li>
</ol></div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">prioritise_targets</span><span class="op">(</span></span>
<span>  results <span class="op">=</span> <span class="fu"><a href="load_example_results.html">load_example_results</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  ctd_list <span class="op">=</span> <span class="fu"><a href="load_example_CTD.html">load_example_ctd</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ctd_DescartesHuman.rds"</span>, <span class="st">"ctd_HumanCellLandscape.rds"</span><span class="op">)</span>,</span>
<span>    multi_dataset <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  phenotype_to_genes <span class="op">=</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/load_phenotype_to_genes.html" class="external-link">load_phenotype_to_genes</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  hpo <span class="op">=</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/get_.html" class="external-link">get_hpo</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  keep_deaths <span class="op">=</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/map_.html" class="external-link">list_deaths</a></span><span class="op">(</span>exclude <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Miscarriage"</span>, <span class="st">"Stillbirth"</span>,</span>
<span>    <span class="st">"Prenatal death"</span><span class="op">)</span>, include_na <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  keep_descendants <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Phenotypic abnormality"</span><span class="op">)</span>,</span>
<span>  keep_ont_levels <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pheno_ndiseases_threshold <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gpt_filters <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  severity_score_gpt_threshold <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  keep_tiers <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  severity_threshold_max <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  info_content_threshold <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  run_prune_ancestors <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  severity_threshold <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pheno_frequency_threshold <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  keep_onsets <span class="op">=</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/map_.html" class="external-link">list_onsets</a></span><span class="op">(</span>include_na <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  effect_var <span class="op">=</span> <span class="st">"logFC"</span>,</span>
<span>  q_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  effect_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  symptom_intersection_threshold <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  keep_celltypes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  evidence_score_threshold <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  keep_chr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">22</span><span class="op">)</span>, <span class="st">"X"</span>, <span class="st">"Y"</span><span class="op">)</span>,</span>
<span>  gene_size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>min <span class="op">=</span> <span class="fl">0</span>, max <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span>,</span>
<span>  gene_frequency_threshold <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  keep_biotypes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  keep_specificity_quantiles <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">30</span>, <span class="fl">40</span><span class="op">)</span>,</span>
<span>  keep_mean_exp_quantiles <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">30</span>, <span class="fl">40</span><span class="op">)</span>,</span>
<span>  sort_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>severity_score_gpt <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, q <span class="op">=</span> <span class="fl">1</span>, logFC <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, specificity <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, mean_exp <span class="op">=</span></span>
<span>    <span class="op">-</span><span class="fl">1</span>, pheno_freq_mean <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, gene_freq_mean <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, width <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>  top_n <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hpo_id"</span><span class="op">)</span>,</span>
<span>  return_report <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-results">results<a class="anchor" aria-label="anchor" href="#arg-results"></a></dt>
<dd><p>The cell type-phenotype enrichment results generated by
<a href="gen_results.html">gen_results</a>
and merged together with <a href="merge_results.html">merge_results</a></p></dd>


<dt id="arg-ctd-list">ctd_list<a class="anchor" aria-label="anchor" href="#arg-ctd-list"></a></dt>
<dd><p>A named list of CellTypeDataset objects each
created with <a href="https://rdrr.io/pkg/EWCE/man/generate_celltype_data.html" class="external-link">generate_celltype_data</a>.</p></dd>


<dt id="arg-phenotype-to-genes">phenotype_to_genes<a class="anchor" aria-label="anchor" href="#arg-phenotype-to-genes"></a></dt>
<dd><p>Output of
<a href="https://rdrr.io/pkg/HPOExplorer/man/load_phenotype_to_genes.html" class="external-link">load_phenotype_to_genes</a> mapping phenotypes
to gene annotations.</p></dd>


<dt id="arg-hpo">hpo<a class="anchor" aria-label="anchor" href="#arg-hpo"></a></dt>
<dd><p>Human Phenotype Ontology object,
loaded from <a href="https://rdrr.io/pkg/KGExplorer/man/get_.html" class="external-link">get_ontology</a>.</p></dd>


<dt id="arg-keep-deaths">keep_deaths<a class="anchor" aria-label="anchor" href="#arg-keep-deaths"></a></dt>
<dd><p>The age of death associated with each HPO ID to keep.
If &gt;1 age of death is associated with the term,
only the earliest age is considered.
See <a href="https://rdrr.io/pkg/HPOExplorer/man/add_.html" class="external-link">add_death</a> for details.</p></dd>


<dt id="arg-keep-descendants">keep_descendants<a class="anchor" aria-label="anchor" href="#arg-keep-descendants"></a></dt>
<dd><p>Terms whose descendants should be kept
(including themselves).
 Set to <code>NULL</code> (default) to skip this filtering step.</p></dd>


<dt id="arg-keep-ont-levels">keep_ont_levels<a class="anchor" aria-label="anchor" href="#arg-keep-ont-levels"></a></dt>
<dd><p>Only keep phenotypes at certain <em>absolute</em>
ontology levels to keep.
See <a href="https://rdrr.io/pkg/HPOExplorer/man/add_.html" class="external-link">add_ont_lvl</a> for details.</p></dd>


<dt id="arg-pheno-ndiseases-threshold">pheno_ndiseases_threshold<a class="anchor" aria-label="anchor" href="#arg-pheno-ndiseases-threshold"></a></dt>
<dd><p>Filter phenotypes by the
maximum number of diseases they are associated with.</p></dd>


<dt id="arg-gpt-filters">gpt_filters<a class="anchor" aria-label="anchor" href="#arg-gpt-filters"></a></dt>
<dd><p>A named list of filters to apply to the GPT annotations.</p></dd>


<dt id="arg-severity-score-gpt-threshold">severity_score_gpt_threshold<a class="anchor" aria-label="anchor" href="#arg-severity-score-gpt-threshold"></a></dt>
<dd><p>The minimum GPT severity score that a
phenotype can have across any disease.</p></dd>


<dt id="arg-keep-tiers">keep_tiers<a class="anchor" aria-label="anchor" href="#arg-keep-tiers"></a></dt>
<dd><p>Tiers from <a href="https://rdrr.io/pkg/HPOExplorer/man/hpo_tiers.html" class="external-link">hpo_tiers</a> to keep.
Include <code>NA</code> if you wish to retain phenotypes that
do not have any Tier assignment.</p></dd>


<dt id="arg-severity-threshold-max">severity_threshold_max<a class="anchor" aria-label="anchor" href="#arg-severity-threshold-max"></a></dt>
<dd><p>The max severity score that a phenotype can
have across any disease.</p></dd>


<dt id="arg-info-content-threshold">info_content_threshold<a class="anchor" aria-label="anchor" href="#arg-info-content-threshold"></a></dt>
<dd><p>Minimum phenotype information content
threshold.</p></dd>


<dt id="arg-run-prune-ancestors">run_prune_ancestors<a class="anchor" aria-label="anchor" href="#arg-run-prune-ancestors"></a></dt>
<dd><p>Prune redundant ancestral terms if any of their
descendants are present. Passes to <a href="https://rdrr.io/pkg/KGExplorer/man/prune_ancestors.html" class="external-link">prune_ancestors</a>.</p></dd>


<dt id="arg-severity-threshold">severity_threshold<a class="anchor" aria-label="anchor" href="#arg-severity-threshold"></a></dt>
<dd><p>Only keep phenotypes with a mean
severity score (averaged across multiple associated diseases) below the
set threshold. The severity score ranges from 1-4 where 1 is the MOST severe.
Include <code>NA</code> if you wish to retain phenotypes that
do not have any severity score.</p></dd>


<dt id="arg-pheno-frequency-threshold">pheno_frequency_threshold<a class="anchor" aria-label="anchor" href="#arg-pheno-frequency-threshold"></a></dt>
<dd><p>Only keep phenotypes with frequency
above the set threshold. Frequency ranges from 0-100 where 100 is
a phenotype that occurs 100% of the time in all associated diseases.
Include <code>NA</code> if you wish to retain phenotypes that
do not have any frequency data.
See <a href="https://rdrr.io/pkg/HPOExplorer/man/add_.html" class="external-link">add_pheno_frequency</a> for details.</p></dd>


<dt id="arg-keep-onsets">keep_onsets<a class="anchor" aria-label="anchor" href="#arg-keep-onsets"></a></dt>
<dd><p>The age of onset associated with each HPO ID to keep.
If &gt;1 age of onset is associated with the term,
only the earliest age is considered.
See <a href="https://rdrr.io/pkg/HPOExplorer/man/add_.html" class="external-link">add_onset</a> for details.</p></dd>


<dt id="arg-effect-var">effect_var<a class="anchor" aria-label="anchor" href="#arg-effect-var"></a></dt>
<dd><p>Name of the effect size column in the <code>results</code>.</p></dd>


<dt id="arg-q-threshold">q_threshold<a class="anchor" aria-label="anchor" href="#arg-q-threshold"></a></dt>
<dd><p>The q value threshold to subset the <code>results</code> by.</p></dd>


<dt id="arg-effect-threshold">effect_threshold<a class="anchor" aria-label="anchor" href="#arg-effect-threshold"></a></dt>
<dd><p>The minimum fold change in specific expression
to subset the <code>results</code> by.</p></dd>


<dt id="arg-symptom-intersection-threshold">symptom_intersection_threshold<a class="anchor" aria-label="anchor" href="#arg-symptom-intersection-threshold"></a></dt>
<dd><p>Minimum proportion of genes
overlapping between a symptom gene list
(phenotype-associated genes in the context of a particular disease)
and the phenotype-cell type association driver genes</p></dd>


<dt id="arg-keep-celltypes">keep_celltypes<a class="anchor" aria-label="anchor" href="#arg-keep-celltypes"></a></dt>
<dd><p>Cell type to keep.</p></dd>


<dt id="arg-evidence-score-threshold">evidence_score_threshold<a class="anchor" aria-label="anchor" href="#arg-evidence-score-threshold"></a></dt>
<dd><p>The minimum threshold of mean
evidence scores of each gene-phenotype association to keep.</p></dd>


<dt id="arg-keep-chr">keep_chr<a class="anchor" aria-label="anchor" href="#arg-keep-chr"></a></dt>
<dd><p>Chromosomes to keep.</p></dd>


<dt id="arg-gene-size">gene_size<a class="anchor" aria-label="anchor" href="#arg-gene-size"></a></dt>
<dd><p>Min/max gene size (important for therapeutics design).</p></dd>


<dt id="arg-gene-frequency-threshold">gene_frequency_threshold<a class="anchor" aria-label="anchor" href="#arg-gene-frequency-threshold"></a></dt>
<dd><p>Only keep genes with frequency
above the set threshold. Frequency ranges from 0-100 where 100 is
a gene that occurs 100% of the time in a given phenotype.
Include <code>NA</code> if you wish to retain genes that
do not have any frequency data.
See <a href="https://rdrr.io/pkg/HPOExplorer/man/add_.html" class="external-link">add_gene_frequency</a> for details.</p></dd>


<dt id="arg-keep-biotypes">keep_biotypes<a class="anchor" aria-label="anchor" href="#arg-keep-biotypes"></a></dt>
<dd><p>Which gene biotypes to keep.
(e.g. "protein_coding", "processed_transcript", "snRNA",
"lincRNA", "snoRNA", "IG_C_gene")</p></dd>


<dt id="arg-keep-specificity-quantiles">keep_specificity_quantiles<a class="anchor" aria-label="anchor" href="#arg-keep-specificity-quantiles"></a></dt>
<dd><p>Which cell type
specificity quantiles to keep (max quantile is 40).</p></dd>


<dt id="arg-keep-mean-exp-quantiles">keep_mean_exp_quantiles<a class="anchor" aria-label="anchor" href="#arg-keep-mean-exp-quantiles"></a></dt>
<dd><p>Which cell type
mean expression quantiles to keep (max quantile is 40).</p></dd>


<dt id="arg-sort-cols">sort_cols<a class="anchor" aria-label="anchor" href="#arg-sort-cols"></a></dt>
<dd><p>How to sort the rows using <a href="https://rdrr.io/pkg/data.table/man/setorder.html" class="external-link">setorderv</a>.
<code>names(sort_cols)</code> will be supplied to the <code>cols=</code> argument
and values will be supplied to the <code>order=</code> argument.</p></dd>


<dt id="arg-top-n">top_n<a class="anchor" aria-label="anchor" href="#arg-top-n"></a></dt>
<dd><p>Top N genes to keep when grouping by <code>group_vars</code>.</p></dd>


<dt id="arg-group-vars">group_vars<a class="anchor" aria-label="anchor" href="#arg-group-vars"></a></dt>
<dd><p>Columns to group by when selecting <code>top_n</code> genes.</p></dd>


<dt id="arg-return-report">return_report<a class="anchor" aria-label="anchor" href="#arg-return-report"></a></dt>
<dd><p>If <code>TRUE</code>, will return a named list containing a
<code>report</code> that shows the number of
phenotypes/celltypes/genes remaining after each filtering step.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A data.table of the prioritised phenotype- and
cell type-specific gene targets.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Term key:<br></p><ul><li><p>Disease:
</p><div class="sourceCode"><pre><code>
A disease defined in the database
OMIM, DECIPHER and/or Orphanet.</code></pre></div></li>
<li><p>Phenotype: A clinical feature associated with one or more diseases.</p></li>
<li><p>Symptom:
</p><div class="sourceCode"><pre><code>
A phenotype within the context of a particular disease.
Within a given phenotype, there may be multiple symptoms with
 partially overlapping genetic mechanisms.</code></pre></div></li>
<li><p>Assocation:
</p><div class="sourceCode"><pre><code>
A cell type-specific enrichment test result conducted
at the disease-level, phenotype-level, or symptom-level.</code></pre></div></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">results</span> <span class="op">=</span> <span class="fu"><a href="load_example_results.html">load_example_results</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="va">q</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">prioritise_targets</span><span class="op">(</span>results<span class="op">=</span><span class="va">results</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading ctd_DescartesHuman.rds</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading ctd_HumanCellLandscape.rds</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2024-12-12</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritising gene targets.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding logFC column.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding HPO names.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating ontology terms to names.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding term definitions.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding information_content scores.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2024-12-12</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding genes and disease IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='start' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 10,938,355 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 9,575 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 8,805 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 201 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 5,092</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering @ q-value &lt;= 0.05</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='q_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 10,938,355 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 9,575 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 8,805 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 201 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 5,092</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering @ logFC &gt;= 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='effect_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 16,235 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 637 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 2,664 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 130 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 2,364</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating phenos with AgeOfDeath.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating ontology terms to ids.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_deaths' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 16,209 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 633 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 2,657 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 130 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 2,362</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding level-2 ancestor to each HPO ID.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ancestor metadata.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Ancestor metadata already present. Use force_new=TRUE to overwrite.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating ontology terms to ids.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Keeping descendants of 1 term(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 18,379 terms remain after filtering.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,198 associations remain after filtering.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_descendants' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 16,198 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 627 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 2,653 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 130 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 2,362</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting absolute ontology level for 19,025 IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_ont_levels' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 16,198 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 627 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 2,653 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 130 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 2,362</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating ontology terms to ids.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2024-12-12</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 383 phenotypes do not have matching HPO IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading in GPT annotations for 16,753 phenotypes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='gpt_filters' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 16,198 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 627 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 2,653 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 130 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 2,362</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='severity_score_gpt_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 5,157 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 258 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1,459 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 89 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 1,457</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='info_content_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 3,605 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 219 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1,040 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 88 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 1,172</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating phenos with onset.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating ontology terms to ids.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_onsets' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 3,605 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 219 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1,040 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 88 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 1,172</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating phenos with Tiers.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_tiers' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 3,605 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 219 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1,040 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 88 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 1,172</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating phenos with modifiers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='severity_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 3,605 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 219 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1,040 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 88 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 1,172</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='severity_threshold_max' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 3,605 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 219 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1,040 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 88 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 1,172</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating phenos with n_diseases</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2024-12-12</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: genes_to_phenotype.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2024-12-12</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype.hpoa</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2024-12-12</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='pheno_ndiseases_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 3,605 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 219 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1,040 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 88 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 1,172</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating phenotype frequencies.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='pheno_frequency_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 3,614 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 219 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1,040 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 88 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 1,172</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_celltypes' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 3,614 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 219 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 1,040 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 88 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 1,172</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting phenos to GRanges.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: ensembldb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering metadata for 1172 unique genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: EnsDb.Hsapiens.v75</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='symptom_gene_overlap' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 3,235 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 219 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 997 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 88 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 1,097</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering by keep_chr.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_chr' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 3,235 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 219 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 997 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 88 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 1,097</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering by gene-disease association evidence.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating gene-disease associations with Evidence Score</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering data from GenCC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing cached file.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Evidence scores for: </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - 10514 diseases </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - 5171 genes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: 2024-12-19</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='evidence_score_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 288 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 70 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 138 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 36 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 139</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering by gene size.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 139 / 139 genes kept.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='gene_size' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 288 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 70 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 138 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 36 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 139</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='keep_biotypes' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 288 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 70 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 138 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 36 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 139</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2024-12-12</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='add_driver_genes' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 202 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 65 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 99 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 33 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 97</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding symptom-level results.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Subsetting results by q_threshold and effect.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 202 associations remain after filtering.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2024-12-12</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping subthreshold symptoms.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='symptom_intersection_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 202 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 65 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 99 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 33 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 97</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotating gene frequencies.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: genes_to_phenotype.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: v2024-12-12</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='gene_frequency_threshold' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 246 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 65 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 99 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 33 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 97</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Pruning ancestors.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 58 / 65 terms were kept after pruning.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='prune_ancestors' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 215 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 58 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 99 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 33 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 97</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sorting rows.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prioritised targets: step='end' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Rows: 215 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Phenotypes: 58 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Diseases: 99 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Cell types: 33 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Genes: 97</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding disease_name and disease_description.</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Robert Gordon-Smith, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

