<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Expression Weighted Celltype Enrichment on Multiple Gene Lists With MultiEWCE • MultiEWCE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Expression Weighted Celltype Enrichment on Multiple Gene Lists With MultiEWCE">
<meta property="og:description" content="MultiEWCE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MultiEWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MultiEWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/MultiEWCE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="MultiEWCE_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Expression Weighted Celltype Enrichment on Multiple Gene Lists With MultiEWCE</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MultiEWCE/blob/master/vignettes/MultiEWCE.Rmd"><code>vignettes/MultiEWCE.Rmd</code></a></small>
      <div class="hidden name"><code>MultiEWCE.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">MultiEWCE</span><span class="op">)</span></code></pre></div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The MultiEWCE package is an extension of the <a href="https://nathanskene.github.io/EWCE/articles/EWCE.html">EWCE</a> package. It is designed to run expression weighted celltype enrichment (EWCE) on multiple gene lists in parallel. The results are then stored both as separate <code>.rds</code> files, one for each individual EWCE analysis, as well as a in a single dataframe containing all the results.</p>
<p>This package is useful in cases where you have a large number of related, but separate, gene lists. In this vignette we will use an example from the Human Phenotype Ontology (HPO). The HPO contains over 9000 clinically relevant phenotypes annotated with lists of genes that have been found to be associated with the particular phenotype.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>This tutorial will require the MultiEWCE package and a few additional packages related to the particular dataset we will be using. Install and load these with the following R code:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="st">"HPOExplorer"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"ovrhuman/HPOExplorer"</span><span class="op">)</span>
<span class="op">}</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="st"><a href="https://github.com/neurogenomics/ewceData">"ewceData"</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"ewceData"</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div id="loading-phenotype-associated-gene-lists-from-the-hpo" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-phenotype-associated-gene-lists-from-the-hpo" class="anchor"></a>Loading Phenotype Associated Gene Lists from the HPO</h1>
<p>The MultiEWCE package requires the gene data to be in a particular format. It must be a data.frame that includes one column of gene list names, and another column of genes. For example:</p>
<table class="table">
<thead><tr class="header">
<th>Phenotype</th>
<th>Gene</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>“Abnormal heart”</td>
<td>gene X</td>
</tr>
<tr class="even">
<td>“Abnormal heart”</td>
<td>gene Y</td>
</tr>
<tr class="odd">
<td>“Poor vision”</td>
<td>gene Z</td>
</tr>
<tr class="even">
<td>“Poor vision”</td>
<td>gene Y</td>
</tr>
<tr class="odd">
<td>“Poor vision”</td>
<td>gene W</td>
</tr>
<tr class="even">
<td>“Short stature”</td>
<td>gene V</td>
</tr>
</tbody>
</table>
<p><em>etc…</em></p>
<p>Now we will get a dataset like this from the HPO.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">gene_data</span> <span class="op">&lt;-</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/load_phenotype_to_genes.html">load_phenotype_to_genes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gene_data</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;           ID                  Phenotype EntrezID     Gene Additional    Source</span>
<span class="co">#&gt; 1 HP:0000002 Abnormality of body height     5073     PARN            orphadata</span>
<span class="co">#&gt; 2 HP:0000002 Abnormality of body height    10084    PQBP1            orphadata</span>
<span class="co">#&gt; 3 HP:0000002 Abnormality of body height    50485 SMARCAL1          -  mim2gene</span>
<span class="co">#&gt; 4 HP:0000002 Abnormality of body height     2317     FLNB            orphadata</span>
<span class="co">#&gt; 5 HP:0000002 Abnormality of body height    79648    MCPH1          -  mim2gene</span>
<span class="co">#&gt; 6 HP:0000002 Abnormality of body height      523  ATP6V1A          -  mim2gene</span>
<span class="co">#&gt;        LinkID</span>
<span class="co">#&gt; 1  ORPHA:1775</span>
<span class="co">#&gt; 2 ORPHA:93950</span>
<span class="co">#&gt; 3 OMIM:242900</span>
<span class="co">#&gt; 4   ORPHA:503</span>
<span class="co">#&gt; 5 OMIM:251200</span>
<span class="co">#&gt; 6 OMIM:617403</span></code></pre></div>
<p>In this example our <strong>gene list names</strong> column is called <code>Phenotype</code> and our column of genes is called <code>Gene</code>. However, different column names can be specified to the MultiEWCE package.</p>
</div>
<div id="setting-up-input-arguments-for-the-gen_results-function" class="section level1">
<h1 class="hasAnchor">
<a href="#setting-up-input-arguments-for-the-gen_results-function" class="anchor"></a>Setting up input arguments for the gen_results function</h1>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Loading CTD file (had to change all rownames to uppercase as the genelists are all upper)</span>

<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_example_CTD.html">load_example_CTD</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">list_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">gene_data</span><span class="op">$</span><span class="va">Phenotype</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>
<span class="va">background_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">gene_data</span><span class="op">$</span><span class="va">Gene</span><span class="op">)</span>
<span class="va">list_name_column</span> <span class="op">&lt;-</span> <span class="st">"Phenotype"</span>
<span class="va">gene_column</span> <span class="op">&lt;-</span> <span class="st">"Gene"</span>
<span class="va">results_dir</span> <span class="op">&lt;-</span> <span class="st">"results"</span>
<span class="va">overwrite_past_analysis</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span>
<span class="va">MergeResults</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span>
<span class="va">reps</span> <span class="op">&lt;-</span> <span class="fl">10</span>
<span class="va">annotLevel</span> <span class="op">&lt;-</span> <span class="fl">1</span>
<span class="va">genelistSpecies</span> <span class="op">&lt;-</span> <span class="st">"human"</span>
<span class="va">sctSpecies</span> <span class="op">&lt;-</span> <span class="st">"human"</span>
<span class="va">cores</span> <span class="op">&lt;-</span> <span class="fl">1</span></code></pre></div>
<div id="ctd" class="section level2">
<h2 class="hasAnchor">
<a href="#ctd" class="anchor"></a>ctd</h2>
<p>The <code>ctd</code> (cell type data) file contains the single cell RNA sequence data that is required for EWCE. for further information about generating a <code>ctd</code> please see the <a href="https://nathanskene.github.io/EWCE/articles/EWCE.html">EWCE</a> documentation. In this example we will use a CTD of human gene expression data, generated from the Descartes Human Cell Atlas. Replace this with your own CTD file.</p>
</div>
<div id="gene_data" class="section level2">
<h2 class="hasAnchor">
<a href="#gene_data" class="anchor"></a>gene_data</h2>
<p>Gene data is the dataframe containin gene list names and genes, in this case we have already loaded it and assigned it to the variable <code>gene_data</code>.</p>
</div>
<div id="list_names" class="section level2">
<h2 class="hasAnchor">
<a href="#list_names" class="anchor"></a>list_names</h2>
<p>This is a character vector containing all the gene list names. This can be obtained from your <code>gene_data</code> as follows. To save time in this example analysis we will only use the first 10 gene lists (<code>[1:10]</code>)</p>
</div>
<div id="background_genes" class="section level2">
<h2 class="hasAnchor">
<a href="#background_genes" class="anchor"></a>background_genes</h2>
<p>This is a character vector of genes to be used as the background genes. See EWCE package docs for more details on background genes.</p>
</div>
<div id="column-names" class="section level2">
<h2 class="hasAnchor">
<a href="#column-names" class="anchor"></a>Column names</h2>
<p><code>list_name_column</code> is the name of the column in gene_data that contains the gene list names and <code>gene_column</code> contains the genes.</p>
</div>
<div id="processing-results-arguments" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-results-arguments" class="anchor"></a>Processing results arguments</h2>
<p>The <code>results_dir</code> argument is the path to the directory where the individual EWCE results will be saved.</p>
<p>The <code>overwrite_past_analysis</code> argument can be set to <code>TRUE</code> or <code>FALSE</code> and states if you want to redo and overwrite analysis of gene lists that have already been saved to the <code>results_dir</code>. Setting this to <code>FALSE</code> is useful in cases where you stopped an analysis midway and would like to carry on from where you left off.</p>
<p><code>MergeResults</code> is another <code>TRUE</code>/<code>FALSE</code> argument. If <code>TRUE</code>, <code>gen_results</code> will also return a single dataframe containing all of the results from all gene lists. A <code>.rds</code> file of these merged results will also be saved to your current working directory in a file called “<code>results_&lt;datetime_of_analysis&gt;.rds</code>” ## Number of cores The <code>cores</code> argument is the number of cores you would like to run in parallel. This is dependent on what is available to you on your computer. In this case we will just run it on one core, no parallelism.</p>
</div>
<div id="ewcebootstrap_enrichment_test-arguments" class="section level2">
<h2 class="hasAnchor">
<a href="#ewcebootstrap_enrichment_test-arguments" class="anchor"></a>EWCE::bootstrap_enrichment_test arguments</h2>
<p>The <code>gen_results</code> function calls the EWCE <code>bootstrap_enrichment_test</code> function. Here we set the input parameters related to this.</p>
<p><code>reps</code> is the number of bootstrap reps to run, for this tutorial we will only do 10 to save time, but typically you would want to do closer to 100,000.</p>
<p><code>annotLevel</code> is the level of cell type specificity to use from the <code>ctd</code> file. <code>genelistSpecies</code> can be “human” or “mouse” and is the species of the genes in the gene lists.</p>
<p><code>sctSpecies</code> is the species of the CTD data, also can be “human” or “mouse”.</p>
</div>
</div>
<div id="run-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#run-analysis" class="anchor"></a>Run analysis</h1>
<p>Now we have set up all our desired inputs, we can run the analysis.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_results</span> <span class="op">&lt;-</span><span class="fu">MultiEWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/gen_results.html">gen_results</a></span><span class="op">(</span><span class="va">ctd</span>,
                           <span class="va">gene_data</span>,
                           <span class="va">list_names</span>,
                           <span class="va">background_genes</span>,
                           <span class="va">list_name_column</span>,
                           <span class="va">gene_column</span>,
                           <span class="va">results_dir</span>,
                           <span class="va">overwrite_past_analysis</span>,
                           <span class="va">reps</span>,
                           <span class="va">annotLevel</span>,
                           <span class="va">genelistSpecies</span>,
                           <span class="va">sctSpecies</span>,
                           <span class="va">cores</span>,
                           <span class="va">MergeResults</span><span class="op">)</span> 
<span class="co">#&gt; [1] "Abnormality of body height"</span>
<span class="co">#&gt; [1] "Multicystic kidney dysplasia"</span>
<span class="co">#&gt; [1] "Autosomal dominant inheritance"</span>
<span class="co">#&gt; [1] "Autosomal recessive inheritance"</span>
<span class="co">#&gt; [1] "Abnormal morphology of female internal genitalia"</span>
<span class="co">#&gt; [1] "Functional abnormality of the bladder"</span>
<span class="co">#&gt; [1] "Recurrent urinary tract infections"</span>
<span class="co">#&gt; [1] "Neurogenic bladder"</span>
<span class="co">#&gt; [1] "Urinary urgency"</span>
<span class="co">#&gt; [1] "Hypoplasia of the uterus"</span></code></pre></div>
</div>
<div id="adjust-for-multiple-comparisons" class="section level1">
<h1 class="hasAnchor">
<a href="#adjust-for-multiple-comparisons" class="anchor"></a>Adjust for multiple comparisons</h1>
<p>Seeing as we have ran EWCE on multiple gene lists, we need to make adjustments for multiple comparisons. We can use Benjamini-Hochberg procedure to false discovery rate adjusted p values (which we will call q values here).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_results</span><span class="op">$</span><span class="va">q</span> <span class="op">&lt;-</span> <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span><span class="op">(</span><span class="va">all_results</span><span class="op">$</span><span class="va">p</span>, method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">all_results</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;                                          CellType annotLevel   p fold_change</span>
<span class="co">#&gt; AFP_ALB_positive_cells     AFP_ALB_positive_cells          1 1.0   0.4546570</span>
<span class="co">#&gt; Acinar_cells                         Acinar_cells          1 1.0   0.6111867</span>
<span class="co">#&gt; Adrenocortical_cells         Adrenocortical_cells          1 0.9   0.9373131</span>
<span class="co">#&gt; Amacrine_cells                     Amacrine_cells          1 1.0   0.6735268</span>
<span class="co">#&gt; Antigen_presenting_cells Antigen_presenting_cells          1 1.0   0.6150362</span>
<span class="co">#&gt; Astrocytes                             Astrocytes          1 0.4   0.9943838</span>
<span class="co">#&gt;                          sd_from_mean</span>
<span class="co">#&gt; AFP_ALB_positive_cells     -5.1535687</span>
<span class="co">#&gt; Acinar_cells               -3.1273970</span>
<span class="co">#&gt; Adrenocortical_cells       -0.6819543</span>
<span class="co">#&gt; Amacrine_cells             -5.0691603</span>
<span class="co">#&gt; Antigen_presenting_cells   -4.8191763</span>
<span class="co">#&gt; Astrocytes                 -0.1655517</span>
<span class="co">#&gt;                                                                 Phenotype q</span>
<span class="co">#&gt; AFP_ALB_positive_cells   Abnormal morphology of female internal genitalia 1</span>
<span class="co">#&gt; Acinar_cells             Abnormal morphology of female internal genitalia 1</span>
<span class="co">#&gt; Adrenocortical_cells     Abnormal morphology of female internal genitalia 1</span>
<span class="co">#&gt; Amacrine_cells           Abnormal morphology of female internal genitalia 1</span>
<span class="co">#&gt; Antigen_presenting_cells Abnormal morphology of female internal genitalia 1</span>
<span class="co">#&gt; Astrocytes               Abnormal morphology of female internal genitalia 1</span></code></pre></div>
</div>
<div id="visualise-the-results" class="section level1">
<h1 class="hasAnchor">
<a href="#visualise-the-results" class="anchor"></a>Visualise the results</h1>
<p>Just as an example, we will create a plot showing the number of significant enrichments per phenotype in the all_results data.frame. We will use q &lt;= 0.05 as the significance threshold.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="va">n_signif</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">p</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">all_results</span><span class="op">$</span><span class="va">Phenotype</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">n_signif</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">n_signif</span>, 
                    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="st">"Phenotype"</span> <span class="op">=</span> <span class="va">p</span>
                               ,<span class="st">"signif_enrichments"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span>
                                 <span class="va">all_results</span><span class="op">$</span><span class="va">q</span><span class="op">[</span><span class="va">all_results</span><span class="op">$</span><span class="va">Phenotype</span> <span class="op">==</span> <span class="va">p</span> <span class="op">&amp;</span>
                                                 <span class="va">all_results</span><span class="op">$</span><span class="va">q</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>


<span class="va">plot1</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">n_signif</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">Phenotype</span>,<span class="va">signif_enrichments</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"Phenotype"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Enrichments (n)"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle<span class="op">=</span><span class="fl">90</span>,vjust <span class="op">=</span> <span class="fl">0.5</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">plot1</span><span class="op">)</span></code></pre></div>
<p><img src="MultiEWCE_files/figure-html/display-1.png" width="672"></p>
</div>
<div id="other-functions-in-multiewce" class="section level1">
<h1 class="hasAnchor">
<a href="#other-functions-in-multiewce" class="anchor"></a>Other functions in MultiEWCE</h1>
<div id="merge_results" class="section level2">
<h2 class="hasAnchor">
<a href="#merge_results" class="anchor"></a>merge_results</h2>
<p>If you have a results directory of individual EWCE results but do not have the merged dataframe of all results, you can call the <code>merge_results</code> function manually. The <code>results_dir</code> argument is the path to your results directory and the <code>list_name_column</code> argument is the name of the column containing gene list names. In this case we used “Phenotype” as this column name when we generated the results.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_results_2</span> <span class="op">&lt;-</span> <span class="fu">MultiEWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/merge_results.html">merge_results</a></span><span class="op">(</span>results_dir <span class="op">=</span> <span class="st">"results"</span>,
                                          list_name_column <span class="op">=</span> <span class="st">"Phenotype"</span><span class="op">)</span></code></pre></div>
</div>
<div id="get_gene_list" class="section level2">
<h2 class="hasAnchor">
<a href="#get_gene_list" class="anchor"></a>get_gene_list</h2>
<p>This function gets a character vector of genes assocaited with a particular gene list name.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">eye_contact_genes</span> <span class="op">&lt;-</span> <span class="fu">MultiEWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/get_gene_list.html">get_gene_list</a></span><span class="op">(</span>list_name <span class="op">=</span> <span class="st">"Poor eye contact"</span>, 
                                      gene_data <span class="op">=</span> <span class="va">gene_data</span>,
                                      list_name_column <span class="op">=</span> <span class="st">"Phenotype"</span>,
                                      gene_column <span class="op">=</span> <span class="st">"Gene"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"5 genes associated with poor eye contact:"</span>,<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">eye_contact_genes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>,collapse <span class="op">=</span> <span class="st">" "</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; 5 genes associated with poor eye contact: TWNK PIGP CDKL5 NDUFAF4 NDUFC2</span></code></pre></div>
</div>
<div id="get_unfinished_list_names" class="section level2">
<h2 class="hasAnchor">
<a href="#get_unfinished_list_names" class="anchor"></a>get_unfinished_list_names</h2>
<p>This function is used to find which gene lists you have not yet analysed</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">unfinished</span> <span class="op">&lt;-</span> <span class="fu">MultiEWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/get_unfinished_list_names.html">get_unfinished_list_names</a></span><span class="op">(</span>list_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">gene_data</span><span class="op">$</span><span class="va">Phenotype</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>,
                                        results_dir <span class="op">=</span> <span class="st">"results"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"Gene lists not yet analysed:\n"</span>,<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">unfinished</span>, collapse <span class="op">=</span> <span class="st">",\n"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Gene lists not yet analysed:</span>
<span class="co">#&gt; Abnormality of the bladder,</span>
<span class="co">#&gt; Bladder diverticulum,</span>
<span class="co">#&gt; Urinary retention,</span>
<span class="co">#&gt; Nocturia,</span>
<span class="co">#&gt; Migrating focal seizure</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Robert Gordon-Smith, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
