<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting Started • MultiEWCE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting Started">
<meta property="og:description" content="MultiEWCE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MultiEWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/MultiEWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/MultiEWCE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting Started</h1>
                        <h4 data-toc-skip class="author"><h4>
<br>
Authors: <i>Robert Gordon-Smith, Brian Schilder, Nathan Skene</i><br>
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
<br>
Vignette updated: <i>Nov-04-2023</i><br>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MultiEWCE/blob/HEAD/vignettes/MultiEWCE.Rmd" class="external-link"><code>vignettes/MultiEWCE.Rmd</code></a></small>
      <div class="hidden name"><code>MultiEWCE.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/MultiEWCE" class="external-link">MultiEWCE</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Registered S3 method overwritten by 'ggtree':</span></span>
<span><span class="co">##   method         from     </span></span>
<span><span class="co">##   fortify.igraph ggnetwork</span></span></code></pre>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <code>MultiEWCE</code> package is an extension of the <a href="https://nathanskene.github.io/EWCE/articles/EWCE.html" class="external-link">EWCE</a>
package. It is designed to run expression weighted celltype enrichment
(<code>EWCE</code>) on multiple gene lists in parallel. The results are
then stored both as separate <em>.rds</em> files, one for each
individual <code>EWCE</code> analysis, as well as a in a single
dataframe containing all the results.</p>
<p>This package is useful in cases where you have a large number of
related, but separate, gene lists. In this vignette we will use an
example from the Human Phenotype Ontology (HPO). The HPO contains over
9000 clinically relevant phenotypes annotated with lists of genes that
have been found to be associated with the particular phenotype.</p>
</div>
<div class="section level2">
<h2 id="loading-phenotype-associated-gene-lists-from-the-hpo">Loading Phenotype Associated Gene Lists from the HPO<a class="anchor" aria-label="anchor" href="#loading-phenotype-associated-gene-lists-from-the-hpo"></a>
</h2>
<p>The MultiEWCE package requires the gene data to be in a particular
format. It must be a data.frame that includes one column of gene list
names, and another column of genes. For example:</p>
<table class="table">
<thead><tr class="header">
<th>Phenotype</th>
<th>Gene</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>“Abnormal heart”</td>
<td>gene X</td>
</tr>
<tr class="even">
<td>“Abnormal heart”</td>
<td>gene Y</td>
</tr>
<tr class="odd">
<td>“Poor vision”</td>
<td>gene Z</td>
</tr>
<tr class="even">
<td>“Poor vision”</td>
<td>gene Y</td>
</tr>
<tr class="odd">
<td>“Poor vision”</td>
<td>gene W</td>
</tr>
<tr class="even">
<td>“Short stature”</td>
<td>gene V</td>
</tr>
</tbody>
</table>
<p><em>etc…</em></p>
<p>Now we will get a dataset like this from the HPO.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_data</span> <span class="op">&lt;-</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/load_phenotype_to_genes.html" class="external-link">load_phenotype_to_genes</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_data</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">hpo_id</th>
<th align="left">hpo_name</th>
<th align="right">ncbi_gene_id</th>
<th align="left">gene_symbol</th>
<th align="left">disease_id</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">HP:0025115</td>
<td align="left">Civatte bodies</td>
<td align="right">64131</td>
<td align="left">XYLT1</td>
<td align="left">OMIM:264800</td>
</tr>
<tr class="even">
<td align="left">HP:0025115</td>
<td align="left">Civatte bodies</td>
<td align="right">64132</td>
<td align="left">XYLT2</td>
<td align="left">OMIM:264800</td>
</tr>
<tr class="odd">
<td align="left">HP:0025115</td>
<td align="left">Civatte bodies</td>
<td align="right">368</td>
<td align="left">ABCC6</td>
<td align="left">OMIM:264800</td>
</tr>
<tr class="even">
<td align="left">HP:0025114</td>
<td align="left">Hypergranulosis</td>
<td align="right">242</td>
<td align="left">ALOX12B</td>
<td align="left">OMIM:242100</td>
</tr>
<tr class="odd">
<td align="left">HP:0025114</td>
<td align="left">Hypergranulosis</td>
<td align="right">59344</td>
<td align="left">ALOXE3</td>
<td align="left">OMIM:242100</td>
</tr>
<tr class="even">
<td align="left">HP:0025114</td>
<td align="left">Hypergranulosis</td>
<td align="right">4014</td>
<td align="left">LORICRIN</td>
<td align="left">OMIM:604117</td>
</tr>
</tbody>
</table>
<p>In this example our <strong>gene list names</strong> column is called
<code>Phenotype</code> and our column of genes is called
<code>Gene</code>. However, different column names can be specified to
the <code>MultiEWCE</code> package.</p>
</div>
<div class="section level2">
<h2 id="setting-up-input-arguments-for-the-gen_results-function">Setting up input arguments for the gen_results function<a class="anchor" aria-label="anchor" href="#setting-up-input-arguments-for-the-gen_results-function"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Loading CTD file </span></span>
<span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_example_CTD.html">load_example_ctd</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">list_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">gene_data</span><span class="op">$</span><span class="va">hpo_name</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span><span class="op">]</span> </span>
<span><span class="va">reps</span> <span class="op">&lt;-</span> <span class="fl">10</span> <span class="co"># in practice would use more reps</span></span>
<span><span class="va">cores</span> <span class="op">&lt;-</span> <span class="fl">1</span> <span class="co"># in practice would use more cores</span></span>
<span><span class="va">save_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">save_dir_tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">save_dir</span>,<span class="st">"results"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="ctd">ctd<a class="anchor" aria-label="anchor" href="#ctd"></a>
</h3>
<p>The <code>ctd</code> (cell type data) file contains the single cell
RNA sequence data that is required for EWCE. for further information
about generating a <code>ctd</code> please see the <a href="https://nathanskene.github.io/EWCE/articles/EWCE.html" class="external-link">EWCE</a>
documentation. In this example we will use a CTD of human gene
expression data, generated from the Descartes Human Cell Atlas. Replace
this with your own CTD file.</p>
</div>
<div class="section level3">
<h3 id="gene_data">gene_data<a class="anchor" aria-label="anchor" href="#gene_data"></a>
</h3>
<p>Gene data is the dataframe containing gene list names and genes, in
this case we have already loaded it and assigned it to the variable
<code>gene_data</code>.</p>
</div>
<div class="section level3">
<h3 id="list_names">list_names<a class="anchor" aria-label="anchor" href="#list_names"></a>
</h3>
<p>This is a character vector containing all the gene list names. This
can be obtained from your <code>gene_data</code> as follows. To save
time in this example analysis we will only use the first 10 gene lists
(<code>[1:10]</code>)</p>
</div>
<div class="section level3">
<h3 id="bg">bg<a class="anchor" aria-label="anchor" href="#bg"></a>
</h3>
<p>This is a character vector of genes to be used as the background
genes. See <code>EWCE</code> package docs for more details on background
genes.</p>
</div>
<div class="section level3">
<h3 id="column-names">Column names<a class="anchor" aria-label="anchor" href="#column-names"></a>
</h3>
<p><code>list_name_column</code> is the name of the column in gene_data
that contains the gene list names and <code>gene_column</code> contains
the genes.</p>
</div>
<div class="section level3">
<h3 id="processing-results-arguments">Processing results arguments<a class="anchor" aria-label="anchor" href="#processing-results-arguments"></a>
</h3>
<p>The <code>save_dir</code> argument is the path to the directory where
the individual EWCE results will be saved.</p>
<p>The <code>force_new</code> argument can be set to <code>TRUE</code>
or <code>FALSE</code> and states if you want to redo and overwrite
analysis of gene lists that have already been saved to the
<code>save_dir</code>. Setting this to <code>FALSE</code> is useful in
cases where you stopped an analysis midway and would like to carry on
from where you left off.</p>
</div>
<div class="section level3">
<h3 id="number-of-cores">Number of cores<a class="anchor" aria-label="anchor" href="#number-of-cores"></a>
</h3>
<p>The <code>cores</code> argument is the number of cores you would like
to run in parallel. This is dependent on what is available to you on
your computer. In this case we will just run it on one core, no
parallelism.</p>
</div>
<div class="section level3">
<h3 id="ewcebootstrap_enrichment_test-arguments">EWCE::bootstrap_enrichment_test arguments<a class="anchor" aria-label="anchor" href="#ewcebootstrap_enrichment_test-arguments"></a>
</h3>
<p>The <code>gen_results</code> function calls the
<code><a href="https://rdrr.io/pkg/EWCE/man/bootstrap_enrichment_test.html" class="external-link">EWCE::bootstrap_enrichment_test</a></code> function. Here we set the
input parameters related to this.</p>
<p><code>reps</code> is the number of bootstrap reps to run, for this
tutorial we will only do 10 to save time, but typically you would want
to do closer to 100,000.</p>
</div>
</div>
<div class="section level2">
<h2 id="run-analysis">Run analysis<a class="anchor" aria-label="anchor" href="#run-analysis"></a>
</h2>
<p>Now we have set up all our desired inputs, we can run the
analysis.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_results</span> <span class="op">&lt;-</span> <span class="fu">MultiEWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/gen_results.html">gen_results</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>                                      gene_data <span class="op">=</span> <span class="va">gene_data</span>,</span>
<span>                                      list_names <span class="op">=</span> <span class="va">list_names</span>, </span>
<span>                                      list_name_column <span class="op">=</span> <span class="st">"hpo_name"</span>,</span>
<span>                                      reps <span class="op">=</span> <span class="va">reps</span>,</span>
<span>                                      cores <span class="op">=</span> <span class="va">cores</span>,</span>
<span>                                      save_dir <span class="op">=</span> <span class="va">save_dir</span>, </span>
<span>                                      save_dir_tmp <span class="op">=</span> <span class="va">save_dir_tmp</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">## Validating gene lists..</span></span></code></pre>
<pre><code><span><span class="co">## 1 / 3 gene lists are valid.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene counts.</span></span></code></pre>
<pre><code><span><span class="co">## Done in: 4.6 seconds.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Saving results ==&gt; /tmp/RtmpHCj4IH/gen_results.rds</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="visualise-the-results">Visualise the results<a class="anchor" aria-label="anchor" href="#visualise-the-results"></a>
</h2>
<p>Just as an example, we will create a plot showing the number of
significant enrichments per phenotype in the all_results data.frame. We
will use q &lt;= 0.05 as the significance threshold.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="co">#### Aggregate results ####</span></span>
<span><span class="va">n_signif</span> <span class="op">&lt;-</span> <span class="va">all_results</span><span class="op">[</span><span class="va">q</span><span class="op">&lt;=</span><span class="fl">0.05</span> <span class="op">&amp;</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">q</span><span class="op">)</span>, <span class="fu">.</span><span class="op">(</span>sig_enrichments <span class="op">=</span> <span class="va">.N</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"hpo_name"</span><span class="op">]</span></span>
<span><span class="co">#### Plot #### </span></span>
<span><span class="va">plot1</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">n_signif</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_wrap.html" class="external-link">str_wrap</a></span><span class="op">(</span><span class="va">hpo_name</span>,width <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>                               y <span class="op">=</span> <span class="va">sig_enrichments</span>,</span>
<span>                               fill <span class="op">=</span> <span class="va">sig_enrichments</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_col</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">"Phenotype"</span>,y<span class="op">=</span><span class="st">"Enrichments (n)"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">methods</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">plot1</span><span class="op">)</span></span></code></pre></div>
<p><img src="MultiEWCE_files/figure-html/display-1.png" width="672"></p>
</div>
<div class="section level2">
<h2 id="other-functions-in-multiewce">Other functions in MultiEWCE<a class="anchor" aria-label="anchor" href="#other-functions-in-multiewce"></a>
</h2>
<div class="section level3">
<h3 id="merge_results">merge_results<a class="anchor" aria-label="anchor" href="#merge_results"></a>
</h3>
<p>If you have a results directory of individual EWCE results but do not
have the merged dataframe of all results, you can call the
<code>merge_results</code> function manually. The <code>save_dir</code>
argument is the path to your results directory and the
<code>list_name_column</code> argument is the name of the column
containing gene list names. In this case we used “Phenotype” as this
column name when we generated the results.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_results_2</span> <span class="op">&lt;-</span> <span class="fu">MultiEWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/merge_results.html">merge_results</a></span><span class="op">(</span>save_dir <span class="op">=</span> <span class="va">save_dir_tmp</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1 results files found.</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="get_gene_list">get_gene_list<a class="anchor" aria-label="anchor" href="#get_gene_list"></a>
</h3>
<p>This function gets a character vector of genes associated with a
particular gene list name.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">phenotypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Scoliosis"</span><span class="op">)</span></span>
<span><span class="va">gene_set</span> <span class="op">&lt;-</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/get_gene_lists.html" class="external-link">get_gene_lists</a></span><span class="op">(</span>phenotypes <span class="op">=</span> <span class="va">phenotypes</span>,</span>
<span>                                        phenotype_to_genes <span class="op">=</span> <span class="va">gene_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Translating all phenotypes to HPO IDs.</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #00BBBB;">ℹ</span> All local files already up-to-date!</span></span></code></pre>
<pre><code><span><span class="co">## + Returning a vector of phenotypes (same order as input).</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">gene_set</span><span class="op">$</span><span class="va">gene_symbol</span><span class="op">)</span><span class="op">)</span>,</span>
<span>          <span class="st">"genes associated with"</span>,<span class="fu"><a href="https://rdrr.io/r/base/shQuote.html" class="external-link">shQuote</a></span><span class="op">(</span><span class="va">phenotypes</span><span class="op">)</span>,<span class="st">":"</span>,</span>
<span>          <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">gene_set</span><span class="op">$</span><span class="va">gene_symbol</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span><span class="op">]</span>,collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1084 genes associated with 'Scoliosis' : PIK3CA, FBXL4, TBX5, TRAPPC4, LZTR1</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="get_unfinished_list_names">get_unfinished_list_names<a class="anchor" aria-label="anchor" href="#get_unfinished_list_names"></a>
</h3>
<p>This function is used to find which gene lists you have not yet
analysed</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_phenotypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">gene_data</span><span class="op">$</span><span class="va">hpo_name</span><span class="op">)</span></span>
<span><span class="va">unfinished</span> <span class="op">&lt;-</span> <span class="fu">MultiEWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/get_unfinished_list_names.html">get_unfinished_list_names</a></span><span class="op">(</span>list_names <span class="op">=</span> <span class="va">all_phenotypes</span>,</span>
<span>                                                   save_dir_tmp <span class="op">=</span> <span class="va">save_dir_tmp</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">methods</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">unfinished</span><span class="op">)</span>,<span class="st">"/"</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">all_phenotypes</span><span class="op">)</span>,</span>
<span>                    <span class="st">"gene lists not yet analysed"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "10968 / 10969 gene lists not yet analysed"</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="run-disease-level-enrichment-tests">Run disease-level enrichment tests<a class="anchor" aria-label="anchor" href="#run-disease-level-enrichment-tests"></a>
</h2>
<p>So far, we’ve iterated over gene list grouped by phenotypes. But we
can also do this at the level of diseases (which are composed of
combinations of phenotypes).</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_data</span> <span class="op">&lt;-</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/load_phenotype_to_genes.html" class="external-link">load_phenotype_to_genes</a></span><span class="op">(</span><span class="st">"genes_to_phenotype.txt"</span><span class="op">)</span></span>
<span><span class="co">#### Filter only to those with &gt;=4 genes ####</span></span>
<span><span class="va">gene_counts</span> <span class="op">&lt;-</span> <span class="va">gene_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>genes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">gene_symbol</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                         by<span class="op">=</span><span class="st">"disease_id"</span><span class="op">]</span><span class="op">[</span><span class="va">genes</span><span class="op">&gt;=</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="va">list_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">gene_counts</span><span class="op">$</span><span class="va">disease_id</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_results</span> <span class="op">&lt;-</span> <span class="fu">MultiEWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/gen_results.html">gen_results</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>                                      gene_data <span class="op">=</span> <span class="va">gene_data</span>,</span>
<span>                                      list_name_column <span class="op">=</span> <span class="st">"disease_id"</span>,</span>
<span>                                      list_names <span class="op">=</span> <span class="va">list_names</span>,</span>
<span>                                      annotLevel <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                      reps <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Results already exist at: /tmp/RtmpHCj4IH/gen_results.rds Use `force_new=TRUE` to overwrite.</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="full-analysis">Full analysis<a class="anchor" aria-label="anchor" href="#full-analysis"></a>
</h2>
<p>Run the following code the replicate the main analysis in the study
described <a href="https://doi.org/10.1101/2023.02.13.23285820" class="external-link">here</a>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_data</span> <span class="op">&lt;-</span> <span class="fu">HPOExplorer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HPOExplorer/man/load_phenotype_to_genes.html" class="external-link">load_phenotype_to_genes</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">gene_data</span><span class="op">[</span>,<span class="va">n_gene</span><span class="op">:=</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">gene_symbol</span><span class="op">)</span><span class="op">)</span>,by<span class="op">=</span><span class="st">"hpo_id"</span><span class="op">]</span></span>
<span><span class="va">gene_data</span> <span class="op">&lt;-</span> <span class="va">gene_data</span><span class="op">[</span><span class="va">n_gene</span><span class="op">&gt;=</span><span class="fl">4</span>,<span class="op">]</span></span>
<span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_example_CTD.html">load_example_ctd</a></span><span class="op">(</span><span class="st">"ctd_DescartesHuman.rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">all_results</span> <span class="op">&lt;-</span> <span class="fu">MultiEWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/gen_results.html">gen_results</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>                                      list_name_column <span class="op">=</span> <span class="st">"hpo_id"</span>, </span>
<span>                                      gene_data <span class="op">=</span> <span class="va">gene_data</span>,  </span>
<span>                                      reps <span class="op">=</span> <span class="fl">100000</span>,</span>
<span>                                      cores <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] data.table_1.14.8 ggplot2_3.4.4     MultiEWCE_0.1.8   BiocStyle_2.30.0 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] later_1.3.1                   bitops_1.0-7                 </span></span>
<span><span class="co">##   [3] ggplotify_0.1.2               GeneOverlap_1.38.0           </span></span>
<span><span class="co">##   [5] filelock_1.0.2                tibble_3.2.1                 </span></span>
<span><span class="co">##   [7] ontologyPlot_1.6              ggnetwork_0.5.12             </span></span>
<span><span class="co">##   [9] graph_1.80.0                  httr2_0.2.3                  </span></span>
<span><span class="co">##  [11] lifecycle_1.0.3               rstatix_0.7.2                </span></span>
<span><span class="co">##  [13] HPOExplorer_0.99.12           rprojroot_2.0.3              </span></span>
<span><span class="co">##  [15] lattice_0.21-9                backports_1.4.1              </span></span>
<span><span class="co">##  [17] magrittr_2.0.3                limma_3.58.1                 </span></span>
<span><span class="co">##  [19] plotly_4.10.3                 sass_0.4.7                   </span></span>
<span><span class="co">##  [21] rmarkdown_2.25                jquerylib_0.1.4              </span></span>
<span><span class="co">##  [23] yaml_2.3.7                    httpuv_1.6.12                </span></span>
<span><span class="co">##  [25] HGNChelper_0.8.1              DBI_1.1.3                    </span></span>
<span><span class="co">##  [27] RColorBrewer_1.1-3            lubridate_1.9.3              </span></span>
<span><span class="co">##  [29] abind_1.4-5                   zlibbioc_1.48.0              </span></span>
<span><span class="co">##  [31] GenomicRanges_1.54.1          purrr_1.0.2                  </span></span>
<span><span class="co">##  [33] BiocGenerics_0.48.1           RCurl_1.98-1.13              </span></span>
<span><span class="co">##  [35] yulab.utils_0.1.0             rappdirs_0.3.3               </span></span>
<span><span class="co">##  [37] GenomeInfoDbData_1.2.11       IRanges_2.36.0               </span></span>
<span><span class="co">##  [39] S4Vectors_0.40.1              gitcreds_0.1.2               </span></span>
<span><span class="co">##  [41] tidytree_0.4.5                piggyback_0.1.5              </span></span>
<span><span class="co">##  [43] pkgdown_2.0.7                 codetools_0.2-19             </span></span>
<span><span class="co">##  [45] DelayedArray_0.28.0           tidyselect_1.2.0             </span></span>
<span><span class="co">##  [47] aplot_0.2.2                   farver_2.1.1                 </span></span>
<span><span class="co">##  [49] matrixStats_1.0.0             stats4_4.3.1                 </span></span>
<span><span class="co">##  [51] BiocFileCache_2.10.1          jsonlite_1.8.7               </span></span>
<span><span class="co">##  [53] ellipsis_0.3.2                systemfonts_1.0.5            </span></span>
<span><span class="co">##  [55] paintmap_1.0                  tools_4.3.1                  </span></span>
<span><span class="co">##  [57] treeio_1.26.0                 ragg_1.2.6                   </span></span>
<span><span class="co">##  [59] Rcpp_1.0.11                   glue_1.6.2                   </span></span>
<span><span class="co">##  [61] SparseArray_1.2.0             xfun_0.41                    </span></span>
<span><span class="co">##  [63] MatrixGenerics_1.14.0         GenomeInfoDb_1.38.0          </span></span>
<span><span class="co">##  [65] RNOmni_1.0.1.2                dplyr_1.1.3                  </span></span>
<span><span class="co">##  [67] withr_2.5.2                   BiocManager_1.30.22          </span></span>
<span><span class="co">##  [69] fastmap_1.1.1                 fansi_1.0.5                  </span></span>
<span><span class="co">##  [71] caTools_1.18.2                digest_0.6.33                </span></span>
<span><span class="co">##  [73] timechange_0.2.0              R6_2.5.1                     </span></span>
<span><span class="co">##  [75] mime_0.12                     gridGraphics_0.5-1           </span></span>
<span><span class="co">##  [77] textshaping_0.3.7             colorspace_2.1-0             </span></span>
<span><span class="co">##  [79] gtools_3.9.4                  RSQLite_2.3.3                </span></span>
<span><span class="co">##  [81] utf8_1.2.4                    tidyr_1.3.0                  </span></span>
<span><span class="co">##  [83] generics_0.1.3                httr_1.4.7                   </span></span>
<span><span class="co">##  [85] htmlwidgets_1.6.2             S4Arrays_1.2.0               </span></span>
<span><span class="co">##  [87] ontologyIndex_2.11            pkgconfig_2.0.3              </span></span>
<span><span class="co">##  [89] gtable_0.3.4                  blob_1.2.4                   </span></span>
<span><span class="co">##  [91] SingleCellExperiment_1.24.0   XVector_0.42.0               </span></span>
<span><span class="co">##  [93] htmltools_0.5.7               carData_3.0-5                </span></span>
<span><span class="co">##  [95] bookdown_0.36                 scales_1.2.1                 </span></span>
<span><span class="co">##  [97] Biobase_2.62.0                png_0.1-8                    </span></span>
<span><span class="co">##  [99] ggfun_0.1.3                   knitr_1.45                   </span></span>
<span><span class="co">## [101] reshape2_1.4.4                coda_0.19-4                  </span></span>
<span><span class="co">## [103] statnet.common_4.9.0          nlme_3.1-163                 </span></span>
<span><span class="co">## [105] curl_5.1.0                    cachem_1.0.8                 </span></span>
<span><span class="co">## [107] stringr_1.5.0                 BiocVersion_3.18.0           </span></span>
<span><span class="co">## [109] KernSmooth_2.23-22            parallel_4.3.1               </span></span>
<span><span class="co">## [111] AnnotationDbi_1.64.1          desc_1.4.2                   </span></span>
<span><span class="co">## [113] pillar_1.9.0                  grid_4.3.1                   </span></span>
<span><span class="co">## [115] vctrs_0.6.4                   gplots_3.1.3                 </span></span>
<span><span class="co">## [117] promises_1.2.1                ggpubr_0.6.0                 </span></span>
<span><span class="co">## [119] car_3.1-2                     dbplyr_2.4.0                 </span></span>
<span><span class="co">## [121] xtable_1.8-4                  Rgraphviz_2.46.0             </span></span>
<span><span class="co">## [123] evaluate_0.23                 orthogene_1.8.0              </span></span>
<span><span class="co">## [125] cli_3.6.1                     compiler_4.3.1               </span></span>
<span><span class="co">## [127] rlang_1.1.2                   crayon_1.5.2                 </span></span>
<span><span class="co">## [129] grr_0.9.5                     ggsignif_0.6.4               </span></span>
<span><span class="co">## [131] labeling_0.4.3                gprofiler2_0.2.2             </span></span>
<span><span class="co">## [133] EWCE_1.11.2                   plyr_1.8.9                   </span></span>
<span><span class="co">## [135] fs_1.6.3                      stringi_1.7.12               </span></span>
<span><span class="co">## [137] BiocParallel_1.36.0           viridisLite_0.4.2            </span></span>
<span><span class="co">## [139] ewceData_1.10.0               network_1.18.1               </span></span>
<span><span class="co">## [141] babelgene_22.9                munsell_0.5.0                </span></span>
<span><span class="co">## [143] Biostrings_2.70.1             gh_1.4.0                     </span></span>
<span><span class="co">## [145] lazyeval_0.2.2                homologene_1.4.68.19.3.27    </span></span>
<span><span class="co">## [147] Matrix_1.6-1.1                ExperimentHub_2.10.0         </span></span>
<span><span class="co">## [149] patchwork_1.1.3               bit64_4.0.5                  </span></span>
<span><span class="co">## [151] KEGGREST_1.42.0               statmod_1.5.0                </span></span>
<span><span class="co">## [153] shiny_1.7.5.1                 highr_0.10                   </span></span>
<span><span class="co">## [155] SummarizedExperiment_1.32.0   interactiveDisplayBase_1.40.0</span></span>
<span><span class="co">## [157] AnnotationHub_3.10.0          broom_1.0.5                  </span></span>
<span><span class="co">## [159] memoise_2.0.1                 bslib_0.5.1                  </span></span>
<span><span class="co">## [161] ggtree_3.10.0                 bit_4.0.5                    </span></span>
<span><span class="co">## [163] ape_5.7-1</span></span></code></pre>
</details><hr>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Robert Gordon-Smith, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
